<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Bibliography notes</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./schedule.html">Course schedule</a></li>
          <li class="active"><a href="./bibliography-notes.html">Bibliography notes</a></li>
          <li><a href="./draft-notes.html">Draft notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Bibliography notes</h1> -->
    
    <p>The idea of using restriction site associated DNA (RAD) markers on a
large scale was first introduced in a paper by <a href="http://genome.cshlp.org/content/17/2/240.long">Miller et al.
(2007)</a> from Eric A.
Johnson's group.</p>
<p>There are important differences between methods based on <strong>restriction
site polymorphism</strong> (like <a href="http://genome.cshlp.org/content/17/2/240.long">Miller et al.
2007</a>) and methods that
use restriction sites as a way to <strong>decrease genomic complexity</strong> before
sequencing.</p>
<p>Below are bibliography notes related to the use of RAD tags.</p>
<h1>*Miller et al. (2007) - Genome Research* (<a href="http://genome.cshlp.org/content/17/2/240.long">link</a>) :major:</h1>
<p>*Rapid and cost-effective polymorphism identification and genotyping
using restriction site associated DNA (RAD) markers*</p>
<ul>
<li>Summary<ul>
<li>Eric A. Johnson's group at the University of Oregon</li>
<li>Definition of RAD tags: "Restriction site associated DNA (RAD)
    tags are a <strong>genome-wide representation of every site of a
    particular restriction enzyme by short DNA tags</strong>"</li>
<li>Here, looking for polymorphism that disrupt restriction sites
    (like RFLP, AFLP) but for many more loci than the previous
    methods.</li>
<li>Not sequencing but use of a micro-array platform</li>
<li>"<strong>RAD markers [...] allow high-throughput, high-resolution
    genotyping in both model and nonmodel systems</strong>"</li>
</ul>
</li>
<li>Interest of RAD tags<ul>
<li>Genetic mapping (recombination mapping, association mapping)
    needs "high-throughput, cost-effective methods to identify and
    genotype polymorphisms"</li>
<li>Previous methods:<ul>
<li>allele-specific oligonucleotide microarrays (thousands of
    SNPs scored per hybridization)</li>
<li>direct hybridization of labelled genomic DNA to short
    oligonucleotide arrays</li>
<li>single feature polymorphisms (SFP) can be discovered and
    typed in a similar way</li>
</ul>
</li>
<li>Those methods need costly preliminary steps (sequencing,
    oligonucleotide synthesis) and are not easily feasible for
    non-model organisms</li>
</ul>
</li>
<li>Principle<ul>
<li>Very good illustration in <a href="http://genome.cshlp.org/content/17/2/240.long#F1">Figure
    1</a>.</li>
<li>Digest - link tags - shear DNA - purify tags - release tags -
    type tags</li>
</ul>
</li>
<li>Demonstration<ul>
<li>Mapping of a breakpoint on a <em>Drosophila</em> chromosome</li>
<li>Identification of the major lateral plate locus in three-spine
    stickleback</li>
</ul>
</li>
<li>Discussion<ul>
<li>Here RAD markers use the <strong>restriction site as the polymorphic
    marker</strong> (no sequencing)</li>
<li>Easy to produce microarray type of genotyping resource</li>
<li>Bulk genotyping is possible</li>
<li>If markers location is determined, genetic mapping is possible</li>
</ul>
</li>
</ul>
<h1>*Miller et al. (2007) - Genome Biology* (<a href="http://genomebiology.com/content/8/6/R105">link</a>) :major:</h1>
<p>*RAD marker microarrays enable rapid mapping of zebrafish mutations*</p>
<ul>
<li>Developed a RAD marker microarray resource for the zebrafish</li>
<li>Bulk segregant approach to map mutations</li>
<li>Assay for individual RAD markers in pooled populations</li>
</ul>
<h2>TODO Read in more details</h2>
<h1>Lewis et al. (2007) - Genetics (<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/">link</a>)</h1>
<p>*High-Density Detection of Restriction-Site-Associated DNA Markers for
Rapid Mapping of Mutated Loci in Neurospora*</p>
<ul>
<li>Micro-array based RAD markers in <em>Neurospora crassa</em></li>
</ul>
<h1>*Baird et al. (2008) - PLoS ONE* (<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376">link</a>) :major:</h1>
<p>*Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers*</p>
<ul>
<li>RAD tags associated with Illumina sequencing for SNP discovery</li>
<li>SNP identified by sequenced polymorphism and disrupted restriction
    sites</li>
<li>Marker density can be tuned by the choice of the enzymes (and their
    number)</li>
<li>Samples can be multiplexed by barcoding</li>
<li>
<p>Useful with or without a reference genome</p>
</li>
<li>
<p><strong>Hohenlohe et al. (2010) - PLoS Genetics</strong>
    (<a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">link</a>)
    :major:</p>
</li>
</ul>
<p>*Population Genomics of Parallel Adaptation in Threespine Stickleback
using Sequenced RAD Tags*</p>
<p>A major paper about sequenced RAD tags.</p>
<ul>
<li>Use natural populations, not lab crosses</li>
<li>Genome scan of nucleotide diversity and differentiation in 5 natural
    populations of threespine stickleback</li>
<li>Illumina-sequenced RAD tags</li>
<li>Identify and type more than 45000 SNPs in each of 100 individuals</li>
<li>Genomic regions with balancing and divergent selection, consistent
    across populations</li>
<li>Identify the genes in regions of interest (regions of parallel
    differentiation across independent populations)</li>
<li>Methods:<ul>
<li>Genome is available</li>
<li>Barcoded sample</li>
<li>Kernel smoothing and permutation testing</li>
</ul>
</li>
<li><strong>Hohenlohe et al. (2011) - Molecular Ecology Resources</strong>
    (<a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract">link</a>)
    :major:</li>
</ul>
<p>*Next-generation RAD sequencing identifies thousands of SNPs for
assessing hybridization between rainbow and westslope cutthroat trout*</p>
<ul>
<li>Nonmodel organism (no genome available)</li>
<li>Salmonid fish, several whole-genome duplication</li>
<li>RAD sequencing to identify SNP loci with fixed allelic differences
    between introduced rainbow trout and native westslope cutthroat
    trout</li>
<li>Use 24 barcode-labelled individuals</li>
<li>
<p>Use Stacks for analysis (genotyping)</p>
</li>
<li>
<p>Etter et al. (2011) - PLoS ONE
    (<a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561">link</a>)</p>
</li>
</ul>
<p>*Local De Novo Assembly of RAD Paired-End Contigs Using Short
Sequencing Reads*</p>
<ul>
<li>Use paired-end sequencing of RAD fragments to build contigs on the
    side which is cut by random shearing</li>
<li>Identify SNP and determine haplotype in threespine stickleback</li>
<li>Produce overlapping contigs of several hundred nucleotides in <em>E.
    coli</em> and threespine stickleback</li>
<li>A circularization step can allow a local assembly of up to 5 kb</li>
</ul>
</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://getpelican.com/">Pelican</a>, a
          static site generator which
          uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>