<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Draft notes</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./schedule.html">Course schedule</a></li>
          <li><a href="./bibliography-notes.html">Bibliography notes</a></li>
          <li class="active"><a href="./draft-notes.html">Draft notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Draft notes</h1> -->
    
    <h1>Course about RAD-tag sequencing - Roscoff 2015</h1>
<h2>Problematic</h2>
<ul>
<li>High coverage</li>
<li>Genome reduction</li>
<li>Non-model species</li>
</ul>
<h2>Presentation of the RAD-tag approach (history)</h2>
<ul>
<li>Protocol</li>
<li>Single or double digest</li>
<li>Single end or paired ends</li>
</ul>
<h2>Use of RAD tags</h2>
<ul>
<li>SNP discovery</li>
<li>Population genetics</li>
<li>Functional genomics</li>
</ul>
<p>#
Practicals
==========</p>
<h2>Teaching objectives</h2>
<ul>
<li>Sequence quality check: trimming, quality control</li>
<li>Stack building (STACKS or other tools, with or without paired-ends
    information)</li>
<li>SNP calling and genotyping (VCFtools)</li>
<li>Allele frequency estimates (individuals or pools?)</li>
<li>Population genetics: comparison of microsats and SNPs trees (cf.
    Anti)</li>
<li>Population genomics: localisation, genome profiling and Gst
    calculation, kernel smoothing</li>
<li>Functional analysis (outside the scope of RAD-tags <em>sensus stricto</em>)</li>
<li>Predicting the number of fragmnts based on published genome sequence
    and restriction sites with python scripts</li>
</ul>
<h2>Available data</h2>
<ul>
<li>Baird 2008: short reads for three-spines and <em>Neurospora crassa</em></li>
<li>Hohenlohe 2010: short reads for three-spines (population genomics of
    parallel evolution)</li>
<li>
<p>Hohenlohe 2011: short reads for rainbow trout and westslope
    cutthroat trout (SNP identification for hybridization diagnostic)</p>
</li>
<li>
<p>Important people and groups</p>
</li>
</ul>
<h2>Three central groups</h2>
<p>To the best of my knowledge, there are three groups which are involved
in the origin of the RAD tags method and which have produced a lot of
the initial papers. They are also producing a lot of the software that
can be used for RAD data analysis. The groups are:
-   Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon):
    <a href="http://molbio.uoregon.edu/johnson/">link</a>
-   William A. Cresko's group (Institute of Ecology and Evolution, U.
    Oregon): <a href="http://creskolab.uoregon.edu/">link</a>. The Cresko lab is
    responsible for the <a href="http://creskolab.uoregon.edu/stacks/">Stacks</a>
    pipeline that can be used to analyse RAD data.
-   Paul A. Hohenlohe (Depts of Biological Science and Statistics, U.
    Idaho): <a href="http://webpages.uidaho.edu/hohenlohe/index.html">link</a>.
    They also have some software and their news page has some
    interesting information about what's going on with RAD.</p>
<h2>People</h2>
<ul>
<li>Miller Michael R.</li>
<li>Johnson Eric A.</li>
<li>Cresko William A.</li>
<li>Baird Nathan A.</li>
<li>
<p>Catchen Julian M.</p>
</li>
<li>
<p>Other interesting papers</p>
</li>
<li>
<p>Hohenlohe 2010, "Using population genomics to detect selection in
    natural populations: key concepts and methodological considerations"
    (<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/">link</a>). An
    interesting review about population genomics.</p>
</li>
</ul>
</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://getpelican.com/">Pelican</a>, a
          static site generator which
          uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>