<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Raw reads processing - Complete workflow</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./practicals.html">Practicals</a></li>
          <li><a href="./bibliography-notes.html">Bibliography notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Raw reads processing - Complete workflow</h1> -->
    
    


<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">Raw read processing</a>
<ul>
<li><a href="#sec-1-1"><span class="todo TODO">TODO</span> add wikipedia links for FASTA, FASTQ and phred score</a></li>
<li><a href="#sec-1-2">FASTA and FASTQ format</a></li>
<li><a href="#sec-1-3">Phred score</a></li>
<li><a href="#sec-1-4">Raw reads files</a>
<ul>
<li><a href="#sec-1-4-1"><span class="todo TODO">TODO</span> Have a look at the raw reads files</a></li>
</ul>
</li>
<li><a href="#sec-1-5">Count the number of reverse reads for each population</a>
<ul>
<li><a href="#sec-1-5-1"><span class="todo TODO">TODO</span> Count the number of reverse reads for HKI population</a></li>
<li><a href="#sec-1-5-2"><span class="todo TODO">TODO</span> Count the number of reverse reads for BYN and RYT populations</a></li>
<li><a href="#sec-1-5-3"><span class="todo TODO">TODO</span> Count the number of reverse reads for the other populations</a></li>
</ul>
</li>
<li><a href="#sec-1-6">Barcode and restriction-site checking and demultiplexing</a>
<ul>
<li><a href="#sec-1-6-1">Detect errors in the barcode and restriction site while demultiplexing</a></li>
<li><a href="#sec-1-6-2"><span class="todo TODO">TODO</span> Extract the reverse reads from SKA</a></li>
<li><a href="#sec-1-6-3"><span class="todo TODO">TODO</span> Extract the reverse reads from LEV</a></li>
<li><a href="#sec-1-6-4"><span class="todo TODO">TODO</span> Get the reads names for each population</a></li>
<li><a href="#sec-1-6-5"><span class="todo TODO">TODO</span> Get the reads with unexpected pattern in <code>s_2_5_sequence.fastq</code></a></li>
<li><a href="#sec-1-6-6"><span class="todo TODO">TODO</span> Complete demultiplexing</a></li>
</ul>
</li>
<li><a href="#sec-1-7">Demultiplexing and sequence sorting</a></li>
<li><a href="#sec-1-8">Quality control</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1">Raw read processing</h2>
<div class="outline-text-2" id="text-1">
<p>
<div class="navLink"><a href="part-one-01-introduction.html">Previous: Introduction</a></div>
</p>
</div>
<div class="outline-3" id="outline-container-sec-1-1">
<h3 id="sec-1-1"><span class="todo TODO">TODO</span> add wikipedia links for FASTA, FASTQ and phred score</h3>
</div>
<div class="outline-3" id="outline-container-sec-1-2">
<h3 id="sec-1-2">FASTA and FASTQ format</h3>
<div class="outline-text-3" id="text-1-2">
<p>
Many high-throughput sequencing methods produce reads in a format called
FASTQ. This format is very similar to the FASTA format, but in addition to the
nucleotide sequence itself it also contains some quality information for each
position.
</p>
<p>
In the FASTA format, each record has two elements: 
</p>
<ul class="org-ul">
<li>the line starting with <code>&gt;</code> contains the record name
</li>
<li>the next line contain the nucleotide sequence
</li>
</ul>
<div class="org-src-container">
<pre class="src src-verb">&gt;myLittleSeq
AATTCCCACAGAATCCCTCGANGGACTGCAAGGCAGCAGCCCATTGCCTAAAAAGGAAGAGTGCACACAGA
</pre>
</div>
<p>
In the FASTQ format, each record has four elements:
</p>
<ul class="org-ul">
<li>the line starting with <code>@</code> contains the record name and precedes the sequence
</li>
<li>the next line is the nucleotide sequence
</li>
<li>the line starting with <code>+</code> repeats the record name and precedes the quality
information
</li>
<li>the next line contains the quality score for each position (one character per
position)
</li>
</ul>
<div class="org-src-container">
<pre class="src src-verb">@myLittleSeq
AATTCCCACAGAATCCCTCGANGGACTGCAAGGCAGCAGCCCATTGCCTAAAAAGGAAGAGTGCACACAGA
+myLittleSeq
BBBBBBB?ABABB@BBB6B?6%6&gt;9B;@BA=;ABAACBABA?=?@AAB&gt;8.(.);@:&gt;BBA7@C@.&lt;(0B8
</pre>
</div>
<p>
The quality score for each position is encoded in one character and can be
converted to a numerical value. This score tells how confident the sequencer is
about the identity of the base at this position of the sequence. One format for
quality score used by Illumina is the Phred score.
</p>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-3">
<h3 id="sec-1-3">Phred score</h3>
<div class="outline-text-3" id="text-1-3">
<p>
To write
</p>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-4">
<h3 id="sec-1-4">Raw reads files</h3>
<div class="outline-text-3" id="text-1-4">
</div><div class="outline-4" id="outline-container-sec-1-4-1">
<h4 id="sec-1-4-1"><span class="todo TODO">TODO</span> Have a look at the raw reads files</h4>
<div class="outline-text-4" id="text-1-4-1">
<p>
The raw reads are separated into several files:
</p>
<div class="org-src-container">
<pre class="src src-verb">s_1_1_sequence.fastq
s_1_2_sequence.fastq
s_2_1_sequence.fastq
s_2_2_sequence.fastq
s_3_1_sequence.fastq
s_3_2_sequence.fastq
s_5_1_sequence.fastq
s_5_2_sequence.fastq
s_6_1_sequence.fastq
s_6_2_sequence.fastq
</pre>
</div>
<p>
The file name format is: <code>s_xxx_yyy_sequence.fastq</code> where:
</p>
<ul class="org-ul">
<li><code>xxx</code> is the sequencing lane (1, 2, 3, 5 or 6 here)
</li>
<li><code>yyy</code> is the read direction (1 for FORWARD reads, 2 for REVERSE reads - we
used paired-end sequencing, remember)
</li>
</ul>
<p>
For example, <code>s_1_1_sequence.fastq</code> and <code>s_1_2_sequence.fastq</code> are the
sequences from lane 1, forward and reverse reads, respectively.
</p>
<p>
Get a look at the file by typing:
</p>
<div class="org-src-container">
<pre class="src src-bash">head s_1_1_sequence.fastq
</pre>
</div>
<pre class="example">
@HWI-EAS418:1:1:3:665#0/1
AATTCATTTACTGTGTAGTTTNTTTGCNGCAAATGAAAAGCAGNCTACATAATGCATAAACAGGCACTGCAAGA
+HWI-EAS418:1:1:3:665#0/1
\`bba`ababbaa`a_[a`a^D^aa`^D[a^S]```URW^b`RD^aabb^^]aabbaSOS__^`bab^a^___a
@HWI-EAS418:1:1:3:1917#0/1
AATTCACATGTGCTCTCTTCCNTTGAGNCGATAAACGCCTCAGAGGTTTTCCTTGTAATCGTGGATGGATGACA
+HWI-EAS418:1:1:3:1917#0/1
`baaa^aa``[\_^a_a__a^DX\^[TDZ\[RL\^[NX\L\HRHVVMM\[]BBBBBBBBBBBBBBBBBBBBBBB
@HWI-EAS418:1:1:3:1141#0/1
AATTCTCTATGGCAACCAATGNACAAATAGTACTAACAGCTTAAAATGTTGGGACACAGTTAAGTGCTCAGCTA
</pre>
<p>
The <code>head</code> command displays the first 10 lines of a file. Can you recognize
which lines contain record names? Record sequences? Quality scores?
</p>
</div>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-5">
<h3 id="sec-1-5">Count the number of reverse reads for each population</h3>
<div class="outline-text-3" id="text-1-5">
<p>
Each population pool was barcoded on one side of the RAD fragment, which
corresponds to the <b>reverse</b> reads here. The barcodes were three letter long,
and were linked to the restriction site of ENZ1. This means that the reverse
reads must start by one of those eight sequences:
</p>
<div class="org-src-container">
<pre class="src src-verb">| Pop | Lane | Barcode | Rev start |
|-----+------+---------+-----------|
| BYN |    1 | CAC     | CACTCC    |
| RYT |    1 | CTT     | CTTTCC    |
| HKI |    2 | TCT     | TCTTCC    |
| PYÖ |    3 | CTT     | CTTTCC    |
| ABB |    3 | TTG     | TTGTCC    |
| SKA |    5 | CAC     | CACTCC    |
| LEV |    5 | TCT     | TCTTCC    |
| POR |    6 | TTG     | TTGTCC    |
</pre>
</div>
<p>
We see that there are two populations pooled in each of lanes 1, 3 and 5, but
only one population in each of lanes 2 and 6.
</p>
</div>
<div class="outline-4" id="outline-container-sec-1-5-1">
<h4 id="sec-1-5-1"><span class="todo TODO">TODO</span> Count the number of reverse reads for HKI population</h4>
<div class="outline-text-4" id="text-1-5-1">
<p>
Let's start with something simple and look at lane 2. There is only one
population, HKI, so all the reverse reads in <code>s_2_2_sequence.fastq</code> should
start with <code>TCTTCC</code>.
</p>
<div class="org-src-container">
<pre class="src src-bash">head s_2_2_sequence.fastq
</pre>
</div>
<pre class="example">
@HWI-EAS418:2:1:2:1978#0/2
TCTTCCCTACACTGCGTGTCGTCTCAATCGCGGGAGCAGCAGTAGACACAGCTAGGGGTGATGTGNGTGTGTGT
+HWI-EAS418:2:1:2:1978#0/2
abbbbbbbbbbbabbb`aabbaba_^`abbaaa_^_aaaa_Z[__\`a`_`aZFY_]YQ^^[_XVDWYXP\T\U
@HWI-EAS418:2:1:4:1179#0/2
TCTTCCCAGCTACGCAGACAATGGCTATCCTTAAAAAGAAAAGTGTGGTTTTCTTACTTTTAACCNTTGAGCCA
+HWI-EAS418:2:1:4:1179#0/2
aabaaaabbaabaaaa`aaa``aaaaa^`aa_aa`aa`_``^`V`\aaX`__]_`a`aaaa``[WD[`ZN^_\\
@HWI-EAS418:2:1:4:1998#0/2
TCTTCCACACCGGGTCAGTCTCACTTTGAAGGAAACTTGGTCCCCTCTAAACTGGAGTTAATCTCNTTGGTTGC
</pre>
<p>
Does the first sequences fulfil this expectation?
</p>
<p>
Of course we do not really want to check manually that each sequence starts
with the correct pattern by visual inspection. One way is to count the total
number of sequences in the file, and another is to count the number of
sequences starting with the correct pattern. Hopefully the numbers match…
</p>
<p>
To count the total number of sequences in the file, we can count its lines with
the command <code>wc -l</code> and then divide by four since each record has four lines:
</p>
<div class="org-src-container">
<pre class="src src-bash">wc -l s_2_2_sequence.fastq
</pre>
</div>
<pre class="example">
1942408 s_2_2_sequence.fastq
</pre>
<p>
How many reads are there in total in this file?
</p>
<p>
Now we can count the number of sequences starting with <code>TCTTCC</code> by using the
<code>grep</code> command and counting the number of lines of its output with <code>wc -l</code>. The
<code>grep</code> command looks for a given pattern in the lines of a file, and output the
matching lines only. The pattern we use for the match is ="<sup>TCTTCC</sup>"<code>. The =^</code>
tells <code>grep</code> that the string should match at the beginning of a line only, not
in the middle of it. The output of <code>grep</code> is then sent to <code>wc -l</code> with a pipe
<code>|</code> so that <code>wc</code> will tell us how many matching lines <code>grep</code> returned. A pipe
sends the output of the first command as an input to the second command.
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^TCTTCC" s_2_2_sequence.fastq | wc -l
</pre>
</div>
<pre class="example">
485602
</pre>
<p>
Do the two numbers match? Is everything as expected? Is our approach robust or
can it fail in some cases?
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-5-2">
<h4 id="sec-1-5-2"><span class="todo TODO">TODO</span> Count the number of reverse reads for BYN and RYT populations</h4>
<div class="outline-text-4" id="text-1-5-2">
<p>
The BYN population shared lane 1 with the RYT population. Reverse reads from
both populations are in <code>s_1_2_sequence.fastq</code>. To count the number of reads
for BYN, we can again use <code>grep</code> and the appropriate pattern:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^CACTCC" s_1_2_sequence.fastq | wc -l
</pre>
</div>
<pre class="example">
187008
</pre>
<p>
And we can do the same for RYT:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^CTTTCC" s_1_2_sequence.fastq | wc -l
</pre>
</div>
<pre class="example">
599202
</pre>
<p>
Finally, we can check that everything makes sense by counting the total number
of reads in <code>s_1_2_sequence.fastq</code>:
</p>
<div class="org-src-container">
<pre class="src src-bash">wc -l s_1_2_sequence.fastq
</pre>
</div>
<pre class="example">
3144840
</pre>
<p>
Do the number match? Does everything make sense?
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-5-3">
<h4 id="sec-1-5-3"><span class="todo TODO">TODO</span> Count the number of reverse reads for the other populations</h4>
<div class="outline-text-4" id="text-1-5-3">
<p>
You can now obtain the number of reverse reads for all the populations. Compare
the number of reads between populations. Is the coverage homogeneous? Can you
explain what you observe?
</p>
</div>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-6">
<h3 id="sec-1-6">Barcode and restriction-site checking and demultiplexing</h3>
<div class="outline-text-3" id="text-1-6">
<p>
In our case, the reads are already quite clean since there is no reads starting
with an unexpected pattern. However, it is possible to have reads that do have
mistakes at the beginning of the sequence, and that should be discarded because
they cannot be assigned to a population for example.
</p>
<p>
In practice, there are methods to correct the barcode for one base mismatch
when the barcodes used for different populations are sufficiently different to
start with (see for example the STACKS pipeline).
</p>
</div>
<div class="outline-4" id="outline-container-sec-1-6-1">
<h4 id="sec-1-6-1">Detect errors in the barcode and restriction site while demultiplexing</h4>
<div class="outline-text-4" id="text-1-6-1">
<p>
Here, we'll just see how to detect those mistakes while demultiplexing,
i.e. while sorting the reads from one file into one file per population. The
file <code>s_5_2_sequence.fastq</code>, which contains the reverse reads for two
populations, was transformed to a low quality file,
<code>s_5_2_sequence.lowQual.fastq</code> in which some nucleotide were randomly changed
(the Phred scores themselves were not touched).
</p>
<p>
Reads in <code>s_5_2_sequence.fastq</code> should start with either <code>CACTCC</code> (SKA
population) or <code>TCTTCC</code> (LEV population). Let's filter the reads from the
modified file, <code>s_5_2_sequence.lowQual.fastq</code> to separate files for each
population, and to a third file for reads that do not match the expected
patterns.
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-6-2">
<h4 id="sec-1-6-2"><span class="todo TODO">TODO</span> Extract the reverse reads from SKA</h4>
<div class="outline-text-4" id="text-1-6-2">
<p>
We use <code>grep</code> again. Since we want to extract full records, not only the
nucleotide sequences that matches the pattern, we use <code>-B 1</code> (extract one line
before the match) and <code>-A 2</code> (extract two lines after the matching line). We
send <code>grep</code> output to a file by using the redirection operator <code>&gt;</code>.
</p>
<p>
In addition, <code>grep</code> will add a <code>--</code> line between groups of contiguous
matches. We do not want to keep it in our output file, so we filter that out
with a reverse <code>grep</code>: a <code>grep</code> call with the <code>-v</code> option which asks <code>grep</code> to
output only the lines that do <b>not</b> match the pattern.
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -B 1 -A 2 "^CACTCC" s_5_2_sequence.lowQual.fastq &gt; SKA-rev.to-clean.fastq
grep -v "^\-" SKA-rev.to-clean.fastq &gt; SKA-rev.final.fastq
</pre>
</div>
<p>
We could do all in one go, without the intermediate <code>SKA-rev.to-clean.fastq</code>
file which contains the <code>--</code> lines, by using a pipe between the two <code>grep</code>
calls:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -B 1 -A 2 "^CACTCC" s_5_2_sequence.lowQual.fastq | grep -v "^\-" &gt; SKA-rev.fastq
</pre>
</div>
<p>
Now we can count the number of sequence in <code>SKA-rev.fastq</code>:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^@" SKA-rev.fastq | wc -l
</pre>
</div>
<pre class="example">
81235
</pre>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-6-3">
<h4 id="sec-1-6-3"><span class="todo TODO">TODO</span> Extract the reverse reads from LEV</h4>
<div class="outline-text-4" id="text-1-6-3">
<p>
Let's do the same for the LEV population:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -B 1 -A 2 "^TCTTCC" s_5_2_sequence.lowQual.fastq | grep -v "^\-" &gt; LEV-rev.fastq
grep "^@" LEV-rev.fastq | wc -l
</pre>
</div>
<pre class="example">
374311
</pre>
<p>
And the total number of initial sequences was:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^@" s_5_2_sequence.lowQual.fastq | wc -l
</pre>
</div>
<pre class="example">
457644
</pre>
<p>
Were there any reads which were not sent to the SKA nor to the LEV file? How
many was this? Can you imagine a way to get those reads to examine their
sequence?
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-6-4">
<h4 id="sec-1-6-4"><span class="todo TODO">TODO</span> Get the reads names for each population</h4>
<div class="outline-text-4" id="text-1-6-4">
<p>
To do things in a more rigorous way, we can actually create one list of reads
names for each population. Those lists will help us to sort the reads later on.
</p>
<p>
As an example, let's get all the names of the reverse reads for BYN. We can do
it in two steps: first we extract the full records of the reverse reads
starting with <code>CACTCC</code> (BYN specific pattern), and then we extract only the
lines containing the reads names. Let's <code>grep</code> again!
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -B 1 -A 2 "^CACTCC" s_1_2_sequence.fastq | grep "^@" &gt; BYN.rev.names
</pre>
</div>
<p>
Prepare a list of reverse reads names for each population, using the
<code>s_2_5_sequence.lowQual.fastq</code> instead of the <code>s_2_5_sequence.fastq</code>.
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-6-5">
<h4 id="sec-1-6-5"><span class="todo TODO">TODO</span> Get the reads with unexpected pattern in <code>s_2_5_sequence.fastq</code></h4>
<div class="outline-text-4" id="text-1-6-5">
<p>
Now you should have a list of the reverse reads for SKA and LEV:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -B 1 -A 2 "^CACTCC" s_5_2_sequence.lowQual.fastq | grep "^@" &gt; SKA.rev.names
grep -B 1 -A 2 "^TCTTCC" s_5_2_sequence.lowQual.fastq | grep "^@" &gt; LEV.rev.names
</pre>
</div>
<p>
We can also prepare the list of all reads in the initial file:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep "^@" s_5_2_sequence.lowQual.fastq &gt; all.names
</pre>
</div>
<p>
Now, by using inverted <code>grep</code> and by giving a file (with the <code>-f</code> option)
containing the patterns to match to <code>grep</code>, we can get the names which were not
assigned to SKA nor to LEV. We also use the <code>-F</code> option to tell <code>grep</code> the
pattern are fix, not regular expressions, to make it faster:
</p>
<div class="org-src-container">
<pre class="src src-bash">grep -F -v -f SKA.rev.names all.names &gt; names.not.in.SKA
grep -F -v -f LEV.rev.names names.not.in.SKA &gt; names.not.in.SKA.nor.LEV
</pre>
</div>
<p>
Great! Now we can use this list of non-matching names to get their actual
sequences and see what were the problems:
</p>
<pre class="example">
grep -F -A 1 -f names.not.in.SKA.nor.LEV s_5_2_sequence.lowQual.fastq &gt; bad.seqs
</pre>
<p>
If we look at <code>bad.seqs</code> with <code>less</code> for example (<code>less bad.seqs</code>), we can see
that there is one mismatch in the first 6 nucleotides.
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-6-6">
<h4 id="sec-1-6-6"><span class="todo TODO">TODO</span> Complete demultiplexing</h4>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-7">
<h3 id="sec-1-7">Demultiplexing and sequence sorting</h3>
<div class="outline-text-3" id="text-1-7">
<p>
Similar to demultiplexing, even though the multiplexing is not really high
here.
</p>
<p>
A few diagnostic plots to see how to see how the data look like. For example,
the number of reads varies between populations. Discussion about issues, the
barcode effects, how to fix that, and similar potential issues within the pools
(how some individuals can have more reads than others, mutations that break the
restriction site).
</p>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-8">
<h3 id="sec-1-8">Quality control</h3>
<div class="outline-text-3" id="text-1-8">
<p>
<div class="navLink"><a href="part-one-03-de-novo-assembly.html">Next: De novo assembly</a></div>
</p>
</div>
</div>
</div>

</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://orgmode.org/">Org mode</a>
          in <a href="http://www.gnu.org/software/emacs/">Emacs</a>
          and <a href="http://getpelican.com/">Pelican</a>, a static site
          generator which uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>