<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>De novo assembly - Complete workflow</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./practicals.html">Practicals</a></li>
          <li><a href="./bibliography-notes.html">Bibliography notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">De novo assembly - Complete workflow</h1> -->
    
    


<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1"><i>De novo</i> assembly</a>
<ul>
<li><a href="#sec-1-1">Assembly into consensus sequences</a>
<ul>
<li><a href="#sec-1-1-1">Concatenate all reads for a given sense</a></li>
<li><a href="#sec-1-1-2"><span class="todo TODO">TODO</span> Run CAP3 (forward reads)</a></li>
<li><a href="#sec-1-1-3"><span class="todo TODO">TODO</span> Run CAP3 (reverse reads)</a></li>
</ul>
</li>
<li><a href="#sec-1-2">Quality filtering of consensus sequences</a>
<ul>
<li><a href="#sec-1-2-1"><span class="todo TODO">TODO</span> Examine read length</a></li>
<li><a href="#sec-1-2-2"><span class="todo TODO">TODO</span> Identify repetitive motives or extremely similar consensus sequences</a></li>
<li><a href="#sec-1-2-3"><span class="todo TODO">TODO</span> Remove microsatellite repeats</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1"><i>De novo</i> assembly</h2>
<div class="outline-text-2" id="text-1">
<p>
<div class="navLink"><a href="part-one-02-raw-reads-processing.html">Previous: Raw reads processing</a></div>
</p>
</div>
<div class="outline-3" id="outline-container-sec-1-1">
<h3 id="sec-1-1">Assembly into consensus sequences</h3>
<div class="outline-text-3" id="text-1-1">
<p>
We choose to go for a <i>de novo</i> assembly since we do not have a genome for this
species. Several tools were tested, and in the end we decided to go with a bit
older tool, CAP3 (<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC310812/">original publication</a>), which can incorporate quality
information in the assembly process.
</p>
<p>
We perform two independent assemblies, one for the forward reads and one for
the reverse reads.
</p>
</div>
<div class="outline-4" id="outline-container-sec-1-1-1">
<h4 id="sec-1-1-1">Concatenate all reads for a given sense</h4>
<div class="outline-text-4" id="text-1-1-1">
<p>
We have to pool all the reads of a given sense together:
</p>
<div class="org-src-container">
<pre class="src src-bash">ls -1
</pre>
</div>
<pre class="example">
ABB.pair.for.fastq.clean
ABB.pair.rev.fastq.clean
BYN.pair.for.fastq.clean
BYN.pair.rev.fastq.clean
HKI.pair.for.fastq.clean
HKI.pair.rev.fastq.clean
LEV.pair.for.fastq.clean
LEV.pair.rev.fastq.clean
POR.pair.for.fastq.clean
POR.pair.rev.fastq.clean
PYO.pair.for.fastq.clean
PYO.pair.rev.fastq.clean
RYT.pair.for.fastq.clean
RYT.pair.rev.fastq.clean
SKA.pair.for.fastq.clean
SKA.pair.rev.fastq.clean
</pre>
<div class="org-src-container">
<pre class="src src-bash">cat *for* &gt; for.fastq
cat *rev* &gt; rev.fastq
</pre>
</div>
<p>
We obtain two large files with all the reads.
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-1-2">
<h4 id="sec-1-1-2"><span class="todo TODO">TODO</span> Run CAP3 (forward reads)</h4>
<div class="outline-text-4" id="text-1-1-2">
<p>
CAP3 takes for input two files, one containing the FASTA sequences and one
containing the corresponding quality scores. We can split the FASTQ files into
those two files using a <a href="resources/split_fasta_qual.py">python script</a> (<code>split_fasta_qual.py</code>):
</p>
<div class="org-src-container">
<pre class="src src-bash">python split_fasta_qual.py for.fastq
python split_fasta_qual.py rev.fastq
</pre>
</div>
<p>
We submit the CAP3 assembly as a job on the ABiMS server. We need to prepare
our <a href="resources/cap3-for-job">job file</a> (<code>cap3-for-job</code>) (see the ABiMS <a href="http://abims.sb-roscoff.fr/resources/cluster/howto">how-to</a> page for more details about
job files and how to submit them):
</p>
<div class="org-src-container">
<pre class="src src-bash">#!/bin/bash
#$ -S /bin/bash
#$ -M matthieu.bruneaux@ens-lyon.org
#$ -V
#$ -m bea
#$ -cwd
cap3 for.fastq.fasta &gt; for.cap3.log
</pre>
</div>
<p>
Here the command is simply <code>cap3 for.fastq.fasta &gt; for.cap3.log</code>. CAP3
automatically gets the quality information from <code>for.fastq.fasta.qual</code>.
</p>
<p>
The job file is submitted to the long queue with:
</p>
<div class="org-src-container">
<pre class="src src-bash">qsub -q long.q cap3-for-job
</pre>
</div>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-1-3">
<h4 id="sec-1-1-3"><span class="todo TODO">TODO</span> Run CAP3 (reverse reads)</h4>
<div class="outline-text-4" id="text-1-1-3">
<p>
Submit a similar assembly for the reverse reads.
</p>
<p>
An alternative approach would have been to map the reads back to the reference
genome. Why did we not use the threespine genome as a reference? 
</p>
<p>
(When mapping back we have to be quite stringent and allow only for example one
or two mismatch, and one unique good quality mapping location. With the
reference genome of a different species, this can be problematic and the
mapping back might be less efficient that creating consensus contigs or
stacks.)
</p>
</div>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-2">
<h3 id="sec-1-2">Quality filtering of consensus sequences</h3>
<div class="outline-text-3" id="text-1-2">
<p>
The <i>de novo</i> assembly with CAP3 can be tuned by changing the parameters from
the default ones. An assembly is likely not to be perfect and the resulting
consensus sequences have to be quality checked before being used for read
mapping.
</p>
</div>
<div class="outline-4" id="outline-container-sec-1-2-1">
<h4 id="sec-1-2-1"><span class="todo TODO">TODO</span> Examine read length</h4>
<div class="outline-text-4" id="text-1-2-1">
<p>
Because of the ddRAD protocol, we expect to produce consensus sequences by
piling up identical or very similar reads into stacks. Since most of the reads
have the same length to start with (xx for forward reads and xx for reverse
reads after cutting the barcode and restriction sites), we expect to obtain
consensus sequences of the same length.
</p>
<p>
We can get the length of the reads by using a homemade <a href="resources/get_read_length.py">python script</a>
(<code>get_read_length.py</code>):
</p>
<div class="org-src-container">
<pre class="src src-bash">python get_read_length.py consensus-for.fasta
</pre>
</div>
<p>
We can plot the read length distribution with R:
</p>
<div class="org-src-container">
<pre class="src src-R"># R script
#
# Plot read length distribution
#------------------------------

d = read.table("consensus-for.fasta.lengths")
plot(hist(d[, 1])
</pre>
</div>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-2-2">
<h4 id="sec-1-2-2"><span class="todo TODO">TODO</span> Identify repetitive motives or extremely similar consensus sequences</h4>
<div class="outline-text-4" id="text-1-2-2">
<p>
CAP3 sometimes produces consensus which contain repetitive motives or very
similar consensus sequences. Those sequences are problematic for the read
mapping back. We perform a blast of the consensus sequences against themselves
to identify sequences which have motives with many matches, or duplicate
consensus sequences.
</p>
<p>
To perform the blast search, we firt prepare a blast database from the
consensus sequences and then we use <code>blastn</code>:
</p>
<div class="org-src-container">
<pre class="src src-bash"># Prepare the blast database

# Perform the blast search (output in tabular format)
</pre>
</div>
<p>
Next, we parse the results to identify consensus sequences to be removed or to
merged together.
</p>
</div>
</div>
<div class="outline-4" id="outline-container-sec-1-2-3">
<h4 id="sec-1-2-3"><span class="todo TODO">TODO</span> Remove microsatellite repeats</h4>
<div class="outline-text-4" id="text-1-2-3">
<p>
The last quality control step is to remove consensus sequences that contains
microsatellite repeats.
</p>
<p>
<div class="navLink"><a href="part-one-04-variant-calling-genotyping.html">Next: Variant calling and genotyping</a></div>
</p>
</div>
</div>
</div>
</div>

</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://orgmode.org/">Org mode</a>
          in <a href="http://www.gnu.org/software/emacs/">Emacs</a>
          and <a href="http://getpelican.com/">Pelican</a>, a static site
          generator which uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>