<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Bibliography notes</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./schedule.html">Course schedule</a></li>
          <li class="active"><a href="./bibliography-notes.html">Bibliography notes</a></li>
          <li><a href="./draft-notes.html">Draft notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Bibliography notes</h1> -->
    
    


<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">Miller et al. (2007) - Genome Research (*)</a></li>
<li><a href="#sec-2">Miller et al. (2007) - Genome Biology (*)</a></li>
<li><a href="#sec-3">Lewis et al. (2007) - Genetics</a></li>
<li><a href="#sec-4">Baird et al. (2008) - PLoS ONE (*)</a></li>
<li><a href="#sec-5">Hohenlohe et al. (2010) - PLoS Genetics (*)</a></li>
<li><a href="#sec-6">Hohenlohe et al. (2011) - Molecular Ecology Resources (*)</a></li>
<li><a href="#sec-7">Etter et al. (2011) - PLoS ONE</a></li>
</ul>
</div>
</div>
<p>
The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by <a href="http://genome.cshlp.org/content/17/2/240.long">Miller et al. (2007)</a> from Eric
A. Johnson's group.
</p>
<p>
There are important differences between methods based on <b>restriction site
polymorphism</b> (like <a href="http://genome.cshlp.org/content/17/2/240.long">Miller et al. 2007</a>) and methods that use restriction sites
as a way to <b>decrease genomic complexity</b> before sequencing.
</p>
<p>
Below are bibliography notes related to the use of RAD tags - (*) denotes
papers of particular interest.
</p>
<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1">Miller et al. (2007) - Genome Research (*)</h2>
<div class="outline-text-2" id="text-1">
<p>
<b><a href="http://genome.cshlp.org/content/17/2/240.long">Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers</a></b>
</p>
<ul class="org-ul">
<li>Summary
<ul class="org-ul">
<li>Eric A. Johnson's group at the University of Oregon
</li>
<li>Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
<b>genome-wide representation of every site of a particular restriction enzyme
by short DNA tags</b>"
</li>
<li>Here, looking for polymorphism that disrupt restriction sites (like RFLP,
AFLP) but for many more loci than the previous methods.
</li>
<li>Not sequencing but use of a micro-array platform
</li>
<li>"<b>RAD markers [â€¦] allow high-throughput, high-resolution genotyping in
both model and nonmodel systems</b>"
</li>
</ul>
</li>
<li>Interest of RAD tags
<ul class="org-ul">
<li>Genetic mapping (recombination mapping, association mapping) needs
"high-throughput, cost-effective methods to identify and genotype
polymorphisms"
</li>
<li>Previous methods:
<ul class="org-ul">
<li>allele-specific oligonucleotide microarrays (thousands of SNPs scored per
hybridization)
</li>
<li>direct hybridization of labelled genomic DNA to short oligonucleotide
arrays
</li>
<li>single feature polymorphisms (SFP) can be discovered and typed in a similar
way
</li>
</ul>
</li>
<li>Those methods need costly preliminary steps (sequencing, oligonucleotide
synthesis) and are not easily feasible for non-model organisms
</li>
</ul>
</li>
<li>Principle
<ul class="org-ul">
<li>Very good illustration in <a href="http://genome.cshlp.org/content/17/2/240.long#F1">Figure 1</a>.
</li>
<li>Digest - link tags - shear DNA - purify tags - release tags - type tags
</li>
</ul>
</li>
<li>Demonstration
<ul class="org-ul">
<li>Mapping of a breakpoint on a <i>Drosophila</i> chromosome
</li>
<li>Identification of the major lateral plate locus in three-spine stickleback
</li>
</ul>
</li>
<li>Discussion
<ul class="org-ul">
<li>Here RAD markers use the <b>restriction site as the polymorphic marker</b> (no
sequencing)
</li>
<li>Easy to produce microarray type of genotyping resource
</li>
<li>Bulk genotyping is possible
</li>
<li>If markers location is determined, genetic mapping is possible
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-2">
<h2 id="sec-2">Miller et al. (2007) - Genome Biology (*)</h2>
<div class="outline-text-2" id="text-2">
<p>
<b><a href="http://genomebiology.com/content/8/6/R105">RAD marker microarrays enable rapid mapping of zebrafish mutations</a></b>
</p>
<ul class="org-ul">
<li>Developed a RAD marker microarray resource for the zebrafish
</li>
<li>Bulk segregant approach to map mutations
</li>
<li>Assay for individual RAD markers in pooled populations
</li>
</ul>
</div>
<ul class="org-ul"><li><span class="todo TODO">TODO</span> Read in more details<br /></li></ul>
</div>
<div class="outline-2" id="outline-container-sec-3">
<h2 id="sec-3">Lewis et al. (2007) - Genetics</h2>
<div class="outline-text-2" id="text-3">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/">High-Density Detection of Restriction-Site-Associated DNA Markers for Rapid
Mapping of Mutated Loci in Neurospora</a></b>
</p>
<ul class="org-ul">
<li>Micro-array based RAD markers in <i>Neurospora crassa</i>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-4">
<h2 id="sec-4">Baird et al. (2008) - PLoS ONE (*)</h2>
<div class="outline-text-2" id="text-4">
<p>
<b><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376">Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers</a></b>
</p>
<ul class="org-ul">
<li>RAD tags associated with Illumina sequencing for SNP discovery
</li>
<li>SNP identified by sequenced polymorphism and disrupted restriction sites
</li>
<li>Marker density can be tuned by the choice of the enzymes (and their number)
</li>
<li>Samples can be multiplexed by barcoding
</li>
<li>Useful with or without a reference genome
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-5">
<h2 id="sec-5">Hohenlohe et al. (2010) - PLoS Genetics (*)</h2>
<div class="outline-text-2" id="text-5">
<p>
<b><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">Population Genomics of Parallel Adaptation in Threespine Stickleback using
Sequenced RAD Tags</a></b>
</p>
<p>
A major paper about sequenced RAD tags.
</p>
<ul class="org-ul">
<li>Use natural populations, not lab crosses
</li>
<li>Genome scan of nucleotide diversity and differentiation in 5 natural
populations of threespine stickleback
</li>
<li>Illumina-sequenced RAD tags
</li>
<li>Identify and type more than 45000 SNPs in each of 100 individuals
</li>
<li>Genomic regions with balancing and divergent selection, consistent across
populations
</li>
<li>Identify the genes in regions of interest (regions of parallel
differentiation across independent populations)
</li>
<li>Methods:
<ul class="org-ul">
<li>Genome is available
</li>
<li>Barcoded sample
</li>
<li>Kernel smoothing and permutation testing
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-6">
<h2 id="sec-6">Hohenlohe et al. (2011) - Molecular Ecology Resources (*)</h2>
<div class="outline-text-2" id="text-6">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract">Next-generation RAD sequencing identifies thousands of SNPs for assessing
 hybridization between rainbow and westslope cutthroat trout</a></b>
</p>
<ul class="org-ul">
<li>Nonmodel organism (no genome available)
</li>
<li>Salmonid fish, several whole-genome duplication
</li>
<li>RAD sequencing to identify SNP loci with fixed allelic differences between
introduced rainbow trout and native westslope cutthroat trout
</li>
<li>Use 24 barcode-labelled individuals
</li>
<li>Use Stacks for analysis (genotyping)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-7">
<h2 id="sec-7">Etter et al. (2011) - PLoS ONE</h2>
<div class="outline-text-2" id="text-7">
<p>
<b><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561">Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads</a></b>
</p>
<ul class="org-ul">
<li>Use paired-end sequencing of RAD fragments to build contigs on the side which
is cut by random shearing
</li>
<li>Identify SNP and determine haplotype in threespine stickleback
</li>
<li>Produce overlapping contigs of several hundred nucleotides in <i>E. coli</i> and
threespine stickleback
</li>
<li>A circularization step can allow a local assembly of up to 5 kb
</li>
</ul>
</div>
</div>

</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://getpelican.com/">Pelican</a>, a
          static site generator which
          uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>