<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Bibliography notes</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./materials.html">Course schedule</a></li>
          <li class="active"><a href="./bibliography-notes.html">Bibliography notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Bibliography notes</h1> -->
    
    


<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">Miller (2007) - Genome Research <b>(§§§)</b></a></li>
<li><a href="#sec-2">Miller (2007) - Genome Biology <b>(§)</b></a></li>
<li><a href="#sec-3">Lewis (2007) - Genetics</a></li>
<li><a href="#sec-4">Baird (2008) - PLoS One <b>(§§§)</b></a></li>
<li><a href="#sec-5">Hohenlohe (2010) - PLoS Genetics <b>(§§§)</b></a></li>
<li><a href="#sec-6">Hohenlohe (2011) - Molecular Ecology Resources <b>(§)</b></a></li>
<li><a href="#sec-7">Emerson (2011) - PNAS <b>(§§)</b></a></li>
<li><a href="#sec-8">Etter (2011) - PLoS One</a></li>
<li><a href="#sec-9">Etter (2011) - Methods in Molecular Biology <b>(Rev)</b></a></li>
<li><a href="#sec-10">Davey (2011) - Briefings in Functional Genomics <b>(Rev)</b></a></li>
<li><a href="#sec-11">Davey (2011) - Nature Reviews Genetics <b>(Rev)</b> <b>(§)</b></a></li>
<li><a href="#sec-12">O'Rourke (2011) - Genetics</a></li>
<li><a href="#sec-13">Pfender (2011) - Theor. Appl. Genet. <b>(§)</b></a></li>
<li><a href="#sec-14">Chong (2011) - Bioinformatics</a></li>
<li><a href="#sec-15">Chutimanitsakun (2011) - BMC Genomics</a></li>
<li><a href="#sec-16">Rowe (2011) - Molecular Ecology <b>(Rev)</b></a></li>
<li><a href="#sec-17">Flight (2012) - Integrative and Comparative Biology <b>(§)</b></a></li>
<li><a href="#sec-18">Hohenlohe (2012) - Phil. Trans. R. Soc. B <b>(§§)</b></a></li>
<li><a href="#sec-19">Houston (2012) - BMC Genomics <b>(§)</b></a></li>
<li><a href="#sec-20">Peterson (2012) - PLoS One <b>(§§§)</b></a></li>
<li><a href="#sec-21">Rubin (2012) - PLoS One <b>(§§)</b></a></li>
<li><a href="#sec-22">Bruneaux, Johnston (2012) - Molecular Ecology <b>(§)</b></a></li>
<li><a href="#sec-23">Miller (2012) - Molecular Ecology</a></li>
<li><a href="#sec-24">Bourgeois (2013) - Molecular Ecology Resources</a></li>
<li><a href="#sec-25">Gautier (2013) - Molecular Ecology <b>(§§§)</b></a></li>
<li><a href="#sec-26">DeFaveri (2013) - Molecular Ecology Resources <b>(§§)</b></a></li>
<li><a href="#sec-27">Hecht (2013) - Molecular Ecology <b>(§§)</b></a></li>
<li><a href="#sec-28">Catchen (2013) - Molecular Ecology <b>(§)</b></a></li>
<li><a href="#sec-29">Hale (2013) - Genes Genomes Genetics <b>(§)</b></a></li>
<li><a href="#sec-30">Catchen STACKS (2013) - Molecular Ecology <b>(§§§)</b></a></li>
<li><a href="#sec-31">Orsini (2013) - Molecular Ecolgy <b>(Rev)</b> <b>(§)</b></a></li>
<li><a href="#sec-32">Hess (2013) - Molecular Ecology <b>(§§)</b></a></li>
<li><a href="#sec-33">Davey (2013) - Molecular Ecology <b>(§)</b></a></li>
<li><a href="#sec-34">McCormack (2013) - Molecular Phylogenetics and Evolution <b>(Rev)</b> <b>(§)</b></a></li>
<li><a href="#sec-35">Nevado (2014) - Molecular Ecology <b>(§)</b></a></li>
<li><a href="#sec-36">Hemmer-Hansen (2014) - Biological Bulletin <b>(Rev)</b></a></li>
<li><a href="#sec-37">Schweyen (2014) - Biological Bulletin <b>(§)</b></a></li>
<li><a href="#sec-38">Stuart (2014) - Science <b>(§)</b></a></li>
<li><a href="#sec-39">Puritz (2014) - PeerJ <b>(§§)</b></a></li>
<li><a href="#sec-40">Andrews and Puritz paper and responses (2014) - Molecular Ecology <b>(Rev)</b> <b>(§§§)</b></a></li>
<li><a href="#sec-41">Heliconius Genome Consortium (2012) - Nature <b>(§)</b></a></li>
<li><a href="#sec-42">Keller (2013) - Molecular Ecology</a>
<ul>
<li><a href="#sec-42-1"><span class="todo TODO">TODO</span> read</a></li>
</ul>
</li>
<li><a href="#sec-43">Parchman (2013) - Molecular Ecology</a>
<ul>
<li><a href="#sec-43-1"><span class="todo TODO">TODO</span> read</a></li>
</ul>
</li>
<li><a href="#sec-44">Heffelfinger (2014) - BMC Genomics <b>(Rev)</b> <b>(§)</b></a></li>
<li><a href="#sec-45">Wu (2010) - BMC Genomics</a></li>
</ul>
</div>
</div>
<p>
Below are bibliography notes related to the use of RAD tags - <b>(§)</b>, <b>(§§)</b> and
<b>(§§§)</b> denotes papers of particular interest, and <b>(Rev)</b> denotes
reviews. Please note that this bibliography is not exhaustive!
</p>
<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1">Miller (2007) - Genome Research <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-1">
<p>
<b><a href="http://genome.cshlp.org/content/17/2/240.long">Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers</a></b>
</p>
<ul class="org-ul">
<li>Summary
<ul class="org-ul">
<li>Eric A. Johnson's group at the University of Oregon
</li>
<li>Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
<b>genome-wide representation of every site of a particular restriction enzyme
by short DNA tags</b>"
</li>
<li>Here, looking for polymorphism that disrupt restriction sites (like RFLP,
AFLP) but for many more loci than the previous methods.
</li>
<li>Not sequencing but use of a micro-array platform
</li>
<li>"<b>RAD markers […] allow high-throughput, high-resolution genotyping in
both model and nonmodel systems</b>"
</li>
</ul>
</li>
<li>Interest of RAD tags
<ul class="org-ul">
<li>Genetic mapping (recombination mapping, association mapping) needs
"high-throughput, cost-effective methods to identify and genotype
polymorphisms"
</li>
<li>Previous methods:
<ul class="org-ul">
<li>allele-specific oligonucleotide microarrays (thousands of SNPs scored per
hybridization)
</li>
<li>direct hybridization of labelled genomic DNA to short oligonucleotide
arrays
</li>
<li>single feature polymorphisms (SFP) can be discovered and typed in a similar
way
</li>
</ul>
</li>
<li>Those methods need costly preliminary steps (sequencing, oligonucleotide
synthesis) and are not easily feasible for non-model organisms
</li>
</ul>
</li>
<li>Principle
<ul class="org-ul">
<li>Very good illustration in <a href="http://genome.cshlp.org/content/17/2/240.long#F1">Figure 1</a>.
</li>
<li>Digest - link tags - shear DNA - purify tags - release tags - type tags
</li>
</ul>
</li>
<li>Demonstration
<ul class="org-ul">
<li>Mapping of a breakpoint on a <i>Drosophila</i> chromosome
</li>
<li>Identification of the major lateral plate locus in three-spine stickleback
</li>
</ul>
</li>
<li>Discussion
<ul class="org-ul">
<li>Here RAD markers use the <b>restriction site as the polymorphic marker</b> (no
sequencing)
</li>
<li>Easy to produce microarray type of genotyping resource
</li>
<li>Bulk genotyping is possible
</li>
<li>If markers location is determined, genetic mapping is possible
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-2">
<h2 id="sec-2">Miller (2007) - Genome Biology <b>(§)</b></h2>
<div class="outline-text-2" id="text-2">
<p>
<b><a href="http://genomebiology.com/content/8/6/R105">RAD marker microarrays enable rapid mapping of zebrafish mutations</a></b>
</p>
<ul class="org-ul">
<li>Developed a RAD marker microarray resource for the zebrafish
</li>
<li>Bulk segregant approach to map mutations
</li>
<li>Assay for individual RAD markers in pooled populations
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-3">
<h2 id="sec-3">Lewis (2007) - Genetics</h2>
<div class="outline-text-2" id="text-3">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/">High-Density Detection of Restriction-Site-Associated DNA Markers for Rapid
Mapping of Mutated Loci in Neurospora</a></b>
</p>
<ul class="org-ul">
<li>Micro-array based RAD markers in <i>Neurospora crassa</i>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-4">
<h2 id="sec-4">Baird (2008) - PLoS One <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-4">
<p>
<b><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376">Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers</a></b>
</p>
<ul class="org-ul">
<li>RAD tags associated with Illumina sequencing for SNP discovery
</li>
<li>SNP identified by sequenced polymorphism and disrupted restriction sites
</li>
<li>Marker density can be tuned by the choice of the enzymes (and their number)
</li>
<li>Samples can be multiplexed by barcoding
</li>
<li>Useful with or without a reference genome
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-5">
<h2 id="sec-5">Hohenlohe (2010) - PLoS Genetics <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-5">
<p>
<b><a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">Population Genomics of Parallel Adaptation in Threespine Stickleback using
Sequenced RAD Tags</a></b>
</p>
<p>
A major paper about sequenced RAD tags.
</p>
<ul class="org-ul">
<li>Use natural populations, not lab crosses
</li>
<li>Genome scan of nucleotide diversity and differentiation in 5 natural
populations of threespine stickleback
</li>
<li>Illumina-sequenced RAD tags
</li>
<li>Identify and type more than 45000 SNPs in each of 100 individuals
</li>
<li>Genomic regions with balancing and divergent selection, consistent across
populations
</li>
<li>Identify the genes in regions of interest (regions of parallel
differentiation across independent populations)
</li>
<li>Methods:
<ul class="org-ul">
<li>Genome is available
</li>
<li>Barcoded sample
</li>
<li>Kernel smoothing and permutation testing
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-6">
<h2 id="sec-6">Hohenlohe (2011) - Molecular Ecology Resources <b>(§)</b></h2>
<div class="outline-text-2" id="text-6">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract">Next-generation RAD sequencing identifies thousands of SNPs for assessing
 hybridization between rainbow and westslope cutthroat trout</a></b>
</p>
<ul class="org-ul">
<li>Nonmodel organism (no genome available)
</li>
<li>Salmonid fish, several whole-genome duplication
</li>
<li>RAD sequencing to identify SNP loci with fixed allelic differences between
introduced rainbow trout and native westslope cutthroat trout
</li>
<li>Use 24 barcode-labelled individuals
</li>
<li>Use Stacks for analysis (genotyping)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-7">
<h2 id="sec-7">Emerson (2011) - PNAS <b>(§§)</b></h2>
<div class="outline-text-2" id="text-7">
<p>
<b><a href="http://www.pnas.org/content/107/37/16196.abstract">Resolving postglacial phylogeography using high-throughput sequencing</a></b>
</p>
<ul class="org-ul">
<li>Fine-scale phylogenetic relationships between recently diverged (from a
refugium after the recession of the Laurentide ice sheet 22000-19000 BP)
populations of pitcher plant mosquito
</li>
<li>Discussed also in <a href="http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0034241">Puritz 2012</a> ("Next-Generation Phylogeography: A Targeted
Approach for Multilocus Sequencing of Non-Model Organisms") in which they use
454 sequencing:
<blockquote>
<p>
"However, few of [the NGS] technologies have been applied in phylogeographic
studies, with the exception of Emerson et al. (42) that used
restriction-site-associated DNA tags (RAD tags) (43,44) to determine the
evolutionary relationship between recently diverged populations of pitcher
plant mosquitos. This methodology can genotype multiple populations at
thousands of SNP loci simultaneously, but have limited ability to survey
large sample sizes within populations because of the cost. For example,
Emerson et al. genotyped 21 different population at ,13,627 different SNPs
but with only 6 individuals per population the authors were forced to
restrict their analysis to 3,741 SNPs ‘‘that were fixed within at least two
populations and were variable among populations’’ and generate one consensus
sequence per locus per population (42). Our targeted 454 sequencing
methodology offers a compromise with the ability to sequence a reasonable
sample size (20 individuals) from one population for 10 different nDNA loci
(,400 bp each) in 1/16th of a plate of 454 sequencing, at lower cost than a
RAD tag analysis."
</p>
</blockquote>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-8">
<h2 id="sec-8">Etter (2011) - PLoS One</h2>
<div class="outline-text-2" id="text-8">
<p>
<b><a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561">Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads</a></b>
</p>
<ul class="org-ul">
<li>Use paired-end sequencing of RAD fragments to build contigs on the side which
is cut by random shearing
</li>
<li>Identify SNP and determine haplotype in threespine stickleback
</li>
<li>Produce overlapping contigs of several hundred nucleotides in <i>E. coli</i> and
threespine stickleback
</li>
<li>A circularization step can allow a local assembly of up to 5 kb
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-9">
<h2 id="sec-9">Etter (2011) - Methods in Molecular Biology <b>(Rev)</b></h2>
<div class="outline-text-2" id="text-9">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3658458/">SNP Discovery and Genotyping for Evolutionary Genetics Using RAD Sequencing</a></b>
</p>
<ul class="org-ul">
<li>Review chapter about RAD genotyping (RAD with Illumina sequencing) or
  <b>RAD-seq</b>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-10">
<h2 id="sec-10">Davey (2011) - Briefings in Functional Genomics <b>(Rev)</b></h2>
<div class="outline-text-2" id="text-10">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3080771/">RADSeq: next-generation population genetics</a></b>
</p>
<ul class="org-ul">
<li>Review about RAD-seq and its use for population genomics
</li>
<li>Interest for non-model organisms, wild populations and ecological population
genomics
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-11">
<h2 id="sec-11">Davey (2011) - Nature Reviews Genetics <b>(Rev)</b> <b>(§)</b></h2>
<div class="outline-text-2" id="text-11">
<p>
<b><a href="http://www.nature.com/nrg/journal/v12/n7/full/nrg3012.html">Genome-wide genetic marker discovery and genotyping using
next-generation sequencing</a></b>
</p>
<ul class="org-ul">
<li>Review about NGS methods that use restriction enzyme digestion to reduce the
genome complexity
</li>
<li>Include reduced-representation sequencing using reduced-representation
libraries (RRLs) or complexity reduction of polymorphic sequences (CRoPS)
</li>
<li>Include RAD-seq
</li>
<li>Include low coverage genotyping
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-12">
<h2 id="sec-12">O'Rourke (2011) - Genetics</h2>
<div class="outline-text-2" id="text-12">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pubmed/21900274">Rapid Mapping and Identification of Mutations in <i>Caenorhabditis elegans</i> by
Restriction Site-Associated DNA Mapping and Genomic Interval Pull-Down Sequencing</a></b>
</p>
<ul class="org-ul">
<li>Forward genetic screens in <i>C. elegans</i> (association between phenotypes and
mutated genes)
</li>
<li>Use RAD-seq for rapid mapping of the mutations
</li>
<li>Use genomic interval pull-down sequencing (GIPS) to capture and sequence
large portions of a mutant genome (ca. Mb)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-13">
<h2 id="sec-13">Pfender (2011) - Theor. Appl. Genet. <b>(§)</b></h2>
<div class="outline-text-2" id="text-13">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pubmed/21344184">Mapping with RAD (restriction-site associated DNA) markers to rapidly identify
QTL for stem rust resistance in <i>Lolium perenne</i></a></b>
</p>
<ul class="org-ul">
<li>Use RAD-seq to produce maps and identify QTL (in association with simple
sequence repeats (SSR) and sequence-tagged sites (STS))
</li>
<li>Reads from parental plants (2) and from F1 (188)
</li>
<li>High variability in the number of reads per F1 (14000 to 935000)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-14">
<h2 id="sec-14">Chong (2011) - Bioinformatics</h2>
<div class="outline-text-2" id="text-14">
<p>
<b><a href="http://bioinformatics.oxfordjournals.org/content/28/21/2732.full">Rainbow: an integrated tool for efficient clustering and
assembling RAD-seq reads</a></b>
</p>
<ul class="org-ul">
<li>Tools to assemble RAD reads
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-15">
<h2 id="sec-15">Chutimanitsakun (2011) - BMC Genomics</h2>
<div class="outline-text-2" id="text-15">
<p>
<b><a href="http://www.biomedcentral.com/1471-2164/12/4">Construction and application for QTL analysis of a Restriction Site Associated
DNA (RAD) linkage map in barley</a></b>
</p>
<ul class="org-ul">
<li>Use RAD protocol to identify markers, build a linkage map and identify QTL
</li>
<li>Low number of RAD markers (530) but obtain results comparable with high
density EST-based SNP map
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-16">
<h2 id="sec-16">Rowe (2011) - Molecular Ecology <b>(Rev)</b></h2>
<div class="outline-text-2" id="text-16">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2011.05197.x/abstract">RAD in the realm of next-generation sequencing technologies</a></b>
</p>
<ul class="org-ul">
<li>Review of RAD as a form of genotyping-by-sequencing
</li>
<li>Applications in genetic mapping, QTL, phylogeography, population genomics
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-17">
<h2 id="sec-17">Flight (2012) - Integrative and Comparative Biology <b>(§)</b></h2>
<div class="outline-text-2" id="text-17">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pubmed/22767487">Genetic Variation in the Acorn Barnacle from Allozymes
to Population Genomics</a></b>
</p>
<ul class="org-ul">
<li>Use allozymes, microsatellites, mtDNA and RADseq
</li>
<li>RADseq on pooled samples (20 individuals per pool) of 3 populations (2 US, 1
UK)
</li>
<li>Use SOAPdenovo assembly
</li>
<li>Fst outlier analysis (using unbiased estimates and empirical distribution,
with the 1% tail of the distribution being the outliers)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-18">
<h2 id="sec-18">Hohenlohe (2012) - Phil. Trans. R. Soc. B <b>(§§)</b></h2>
<div class="outline-text-2" id="text-18">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pubmed/22201169">Extensive linkage disequilibrium and parallel adaptive divergence across
threespine stickleback genomes</a></b>
</p>
<ul class="org-ul">
<li>Study patterns of local and long-distance linkage desequilibrium (LD) across
oceanic and freshwater populations of threespine stickleback
</li>
<li>Look for association between LD and signatures of divergent selection
</li>
<li>Assess the role of recombination rate variation in generating LD patterns
</li>
<li>Reuse RAD-seq data from Hohenlohe 2010 (PLoS Genetics) and also generate new
data with a modified RAD-seq protocol on 87 F2 individuals and the parents
and F1 individuals
</li>
<li>Align raw reads to reference genome with Bowtie
</li>
<li>Use likelihood methods to infer the genotype at each nucleotide
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-19">
<h2 id="sec-19">Houston (2012) - BMC Genomics <b>(§)</b></h2>
<div class="outline-text-2" id="text-19">
<p>
<b><a href="http://www.biomedcentral.com/1471-2164/13/244">Characterisation of QTL-linked and genome-wide restriction site-associated DNA
(RAD) markers in farmed Atlantic salmon</a></b>
</p>
<ul class="org-ul">
<li>RAD-seq on two Atlantic salmon families
</li>
<li>Offspring were classified into resistant and susceptible to a disease, and
for each family DNA from the parents and 7 offsprings of each phenotype was
sequenced by RAD-seq in multiplexed pools
</li>
<li>Use of pedigreed samples allowed to distinguish segregating SNPs from
putative paralogous sequences (recent genome duplication in salmonids)
</li>
<li>50 SNPs linked to QTL and a subset used for high-throughput genotyping across
large commercial populations of disease-challenged fry
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-20">
<h2 id="sec-20">Peterson (2012) - PLoS One <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-20">
<p>
<b><a href="http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0037135">Double Digest RADseq: An Inexpensive Method for <i>De Novo</i> SNP Discovery and
Genotyping in Model and Non-Model Species</a></b>
</p>
<ul class="org-ul">
<li>Introduce the double-digest method
</li>
<li>Combinatorial barcoding for larger number of individuals per multiplex
</li>
<li>Detailed pipeline of reads analysis
</li>
<li>Also compare the observed size distribution with <i>in silico</i> digestion and
statistical models of size selection
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-21">
<h2 id="sec-21">Rubin (2012) - PLoS One <b>(§§)</b></h2>
<div class="outline-text-2" id="text-21">
<p>
<b><a href="http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0033394">Inferring Phylogenies from RAD Sequence Data</a></b>
</p>
<ul class="org-ul">
<li>Simulate RAD-seq data collection for interspecific phylogeny reconstruction
in <i>Drosophila</i>, mammals and yeast
</li>
<li>Develop a workflow to test if informative data can be gained from RAD-seq and
used for accurate reconstruction of "known" phylogenies
</li>
<li>RAD-seq seems promising for phylogenies in younger clades (e.g. 60 My vs 100
My) in which enough orthologous restriction sites are retained across species
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-22">
<h2 id="sec-22">Bruneaux, Johnston (2012) - Molecular Ecology <b>(§)</b></h2>
<div class="outline-text-2" id="text-22">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pubmed/22943747">Molecular evolutionary and population genomic analysis of the nine-spined
stickleback using a modified restriction-site-associated DNA tag approach</a></b>
</p>
<ul class="org-ul">
<li>Double digest RAD-seq on pooled samples of ninespine sticklebacks
</li>
<li>Mapping back to the related threespine stickleback genome
</li>
<li>Issues with low coverage per population, resulting in data being pooled per
ecotype
</li>
<li>Kernel smoothing of differentiation values along the genome
</li>
<li>Permutation testing and gene ontology enrichment test
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-23">
<h2 id="sec-23">Miller (2012) - Molecular Ecology</h2>
<div class="outline-text-2" id="text-23">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1365-294X.2011.05305.x/abstract">A conserved haplotype controls parallel adaptation in geographically distant
salmonid populations</a></b>
</p>
<ul class="org-ul">
<li>Laboratory crosses and NGS to study the genetic architecture of parallel
adaptation of rapid development rates in two salmon populations
</li>
<li>Find parallel genetic mechanism and conservation of haplotype
</li>
<li>Use custom Perl scripts to process the data and Novoalign for alignment
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-24">
<h2 id="sec-24">Bourgeois (2013) - Molecular Ecology Resources</h2>
<div class="outline-text-2" id="text-24">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12137/pdf">Mass production of SNP markers in a nonmodel passerine bird through RAD
sequencing and contig mapping to the zebra finch genome</a></b>
</p>
<ul class="org-ul">
<li>RAD-seq with Illumina HiSeq2000 on nonmodel passerine bird (paired ends)
</li>
<li>6 pools of 18-25 individuals on one lane
</li>
<li>600000 contigs, of which 386000 mapped back to zebra finch genome
</li>
<li>80000 SNPs mapped
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-25">
<h2 id="sec-25">Gautier (2013) - Molecular Ecology <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-25">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12360/abstract">Estimation of population allele frequencies from next-generation sequencing
data: pool-versus individual-based genotyping</a></b>
</p>
<ul class="org-ul">
<li>Mathematical analysis showing that next-generation sequencing methods applied
on DNA pools from diploid individuals give at least the same accuracy as
individual-based analyses for SNPs
</li>
<li>Remains true when unequal contributions of each individual to the final pool
</li>
<li>Introduce effective pool size and Bayesian model to estimate it
</li>
<li>Application provided to assess the accuracy of allele frequency estimates
</li>
<li>Illustration with theoretical examples and real data sets
</li>
<li>"[NGS of DNA pools] allows comparison of genome-wide patterns of genetic
variation for large numbers of individuals in multiple populations"
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-26">
<h2 id="sec-26">DeFaveri (2013) - Molecular Ecology Resources <b>(§§)</b></h2>
<div class="outline-text-2" id="text-26">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/1755-0998.12071/full">Characterizing genic and nongenic molecular markers: comparison of
 microsatellites and SNPs</a></b>
</p>
<ul class="org-ul">
<li>Compares patterns of genetic diversity and divergence on various geographical
scales for SNP vs microsatellites and genic vs non-genic regions
</li>
<li>Marker type and location affec estimates of diversity and divergence
</li>
<li>Between-lineage divergence higher with SNP but within-lineage divergence
similar for both marker types
</li>
<li>Broad-scale population structure resolved by both markers, but only
microsatellites can delimit fine-scale population-structuring
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-27">
<h2 id="sec-27">Hecht (2013) - Molecular Ecology <b>(§§)</b></h2>
<div class="outline-text-2" id="text-27">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12082/abstract">Genome-wide association reveals genetic basis for the propensity to migrate in
 wild populations of rainbow and steelhead trout</a></b>
</p>
<ul class="org-ul">
<li>RAD-seq in two wild populations of migratory steelhead and resident rainbow
trout
</li>
<li>genome-wide association study using 189 individuals, uniquely barcoded and
distributed in five libraries, single reads, 100 bp
</li>
<li>504 SNP markers associated with propensity to migrate both within and between
populations
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-28">
<h2 id="sec-28">Catchen (2013) - Molecular Ecology <b>(§)</b></h2>
<div class="outline-text-2" id="text-28">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12330/abstract">The population structure and recent colonization history of Oregon threespine
 stickleback determined using restriction-site associated DNA-sequencing</a></b>
</p>
<ul class="org-ul">
<li>Study with populations which predate the last glacial maximum
</li>
<li>Use the STRUCTURE program and reconstruct phylogeographic history
</li>
<li>Results support the a break between coastal and inland populations,
introgressive hybridization in coastal populations and a recent expansion in
central Oregon
</li>
<li>"Our results exhibit the power of next-generation sequencing genomic
approaches such as RAD-seq to identify both historical population structure
and recent colonization history"
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-29">
<h2 id="sec-29">Hale (2013) - Genes Genomes Genetics <b>(§)</b></h2>
<div class="outline-text-2" id="text-29">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3737167/">Evaluating Adaptive Divergence Between Migratory and Nonmigratory Ecotypes of
 a Salmonid Fish, <i>Oncorhynchus mykiss</i></a></b>
</p>
<ul class="org-ul">
<li>Association population genomics and association approaches enables to reveal
the genetic basis of variable phenotypic traits, including in nonmodel
species, if population structure does not overlap with the phenotypic
differences (cf. Susan and Tutku paper about sea-wintering in Atlantic
salmon)
</li>
<li>RAD-seq on rainbow trout to study the divergence between migratory and
non-migratory ecotypes
</li>
<li>Can map some SNP markers using previous linkage maps that used RAD tags
</li>
<li>Some SNPs fall into regions which contain QTL for migratory-related traits
</li>
<li>Also use annotation of genome regions linked to the significant SNPs
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-30">
<h2 id="sec-30">Catchen STACKS (2013) - Molecular Ecology <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-30">
<p>
<b><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3936987/">Stacks: an analysis tool set for population genomics</a></b>
</p>
<ul class="org-ul">
<li>STACKS package: genotype-by-sequencing, SNP-by-SNP analysis, sliding window
across a genome
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-31">
<h2 id="sec-31">Orsini (2013) - Molecular Ecolgy <b>(Rev)</b> <b>(§)</b></h2>
<div class="outline-text-2" id="text-31">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12177/abstract">Evolutionary Ecological Genomics</a></b>
</p>
<p>
Review about evolutionary ecological genomics, initiated at the ESEB
meeting 2011. It mentions RAD among other methods.
</p>
</div>
</div>
<div class="outline-2" id="outline-container-sec-32">
<h2 id="sec-32">Hess (2013) - Molecular Ecology <b>(§§)</b></h2>
<div class="outline-text-2" id="text-32">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12150/abstract">Population genomics of Pacific lamprey: adaptive variation in a highly
 dispersive species</a></b>
</p>
<ul class="org-ul">
<li>No strict homing behaviour but genetic heterogeneity and morphological and
behavioural diversity
</li>
<li>RAD-seq from 518 individuals (7 libraries of 96 individually barcoded
samples)
</li>
<li>4439 SNPs, 162 potentially adaptive as detected by outlier tests (formed four
groups of linked loci)
</li>
<li>MATSAM software to associate them with various traits or geography
</li>
<li>some SNPs aligned with gene region using the genome browser available for sea
lamprey
</li>
<li>Introduction: present the issues with Fst and population structure, refer to
Bierne 2011, Le Corre and Kremer 2012, Excoffier and Lischer 2010, Narum and
Hess 2011.
</li>
<li>"Genomic scans using highly dispersive species may be one way to overcome
this challenge, because minimal population structure provides a less noisy
neutral back- ground of genetic divergence upon which adaptive markers can be
readily distinguished as outliers (Pe′rez- Figueroa et al. 2010)."
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-33">
<h2 id="sec-33">Davey (2013) - Molecular Ecology <b>(§)</b></h2>
<div class="outline-text-2" id="text-33">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12084/full">Special features of RAD Sequencing data: implications for genotyping</a></b>
</p>
<ul class="org-ul">
<li>RAD-seq associated with sequencing-by-synthesis method: stochastic count
data, requires sensitive analysis to develop and genotype markers accurately
</li>
<li>Exist several sources of bias not addressed by current genotyping tools
(fragment bias, restriction site heterozygosity and PCR GC cnotent bias)
</li>
<li>Study the effect of those biases on current analysis tools
</li>
<li>Biases must be taken into account, but most RAD loci can be accurately
genotyped by existing tools
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-34">
<h2 id="sec-34">McCormack (2013) - Molecular Phylogenetics and Evolution <b>(Rev)</b> <b>(§)</b></h2>
<div class="outline-text-2" id="text-34">
<p>
<b><a href="http://www.sciencedirect.com/science/article/pii/S1055790311005203">Applications of next-generation sequencing to phylogeography and
 phylogenetics</a></b>
</p>
<ul class="org-ul">
<li>Discuss methods of genome complexity reduction for phylogeography and phylogenetics
</li>
<li>Discuss which tools to use and how to analyze this kind of data
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-35">
<h2 id="sec-35">Nevado (2014) - Molecular Ecology <b>(§)</b></h2>
<div class="outline-text-2" id="text-35">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12693/abstract">Resequencing studies of nonmodel organisms using closely related reference
 genomes: optimal experimental designs and bioinformatics approaches for population genomics</a></b>
</p>
<ul class="org-ul">
<li>Guidelines for estimating variability and neutrality tests from NGS data on
nonmodel species
</li>
<li>Compare performances of genotype calling, genotype-haplotype calling and
direct estimation
</li>
<li>In the end, "[s]tudies without species-specific reference genome should thus
aim for low read depth and avoid genotype calling whenever individual
genotypes are not essential. Otherwise, aiming for moderate to high depth at
the expense of number of individuals, and using genotype–haplotype calling,
is recommended."
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-36">
<h2 id="sec-36">Hemmer-Hansen (2014) - Biological Bulletin <b>(Rev)</b></h2>
<div class="outline-text-2" id="text-36">
<p>
<b><a href="http://www.biolbull.org/content/227/2/117.long">Population Genomics of Marine Fishes: Next-Generation Prospects and Challenges
</a></b>
</p>
<ul class="org-ul">
<li>Review about the methods and challenges for population genomics in fishes
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-37">
<h2 id="sec-37">Schweyen (2014) - Biological Bulletin <b>(§)</b></h2>
<div class="outline-text-2" id="text-37">
<p>
<b><a href="http://www.biolbull.org/content/227/2/146.long">Detection and Removal of PCR Duplicates in Population Genomic ddRAD Studies by
 Addition of a Degenerate Base Region (DBR) in Sequencing Adapters </a></b>
</p>
<ul class="org-ul">
<li>Present the issue of PCR duplicates in the ddRAD protocol (in paired-ends but
single digest, the PCR duplicates can be detected with the 2nd read)
</li>
<li>Propose an approach with a degenerate base region in the sequencing adapter
to identify and remove the PCR duplicates
</li>
<li>In a test, 34% of a ddRAD library were PCR duplicates distributed over 20% of
the loci
</li>
<li>Disproportionate amount of PCR duplicates in 5% of the loci
</li>
<li>Not a problem for general parameter inference, but outlier loci detetion
could be improved by ddRAD
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-38">
<h2 id="sec-38">Stuart (2014) - Science <b>(§)</b></h2>
<div class="outline-text-2" id="text-38">
<p>
<b><a href="http://www.sciencemag.org/content/346/6208/463.abstract">Rapid evolution of a native species following invasion by a congener</a></b>
</p>
<ul class="org-ul">
<li>Detailed section in Methods (supplementary material) about the RAD-seq part
</li>
<li>6 nt barcodes to multiplex 192 individuals per lane (total 384 individuals),
single-end sequencing Illumina HiSeq
</li>
<li>Demulitplex the reads, filter barcode and correct restriction site with
STACKS
</li>
<li>Align reads against a reference genome with Bowtie2 (discard reads which map
to more than one location)
</li>
<li>Diploid genotypes called using a maximum likelihood approach (Catchen 2013
Stacks, Hohenlohe 2010 parallel evolution sticklebacks)
</li>
<li>Neighbor-joining phylogenetic network with SplitsTree4
</li>
<li>Pairwise Fst between populations 
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-39">
<h2 id="sec-39">Puritz (2014) - PeerJ <b>(§§)</b></h2>
<div class="outline-text-2" id="text-39">
<p>
<b><a href="https://peerj.com/articles/431/"><i>dDocent</i>: a RADseq, variant-calling pipeline designed for population genomics
of non-model organisms</a></b>
</p>
<ul class="org-ul">
<li>Pipeline for variant calling using paired-end RAD-seq data
</li>
<li>Relies on existing software (one version on FreeBayes, one on GATK)
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-40">
<h2 id="sec-40">Andrews and Puritz paper and responses (2014) - Molecular Ecology <b>(Rev)</b> <b>(§§§)</b></h2>
<div class="outline-text-2" id="text-40">
<p>
A set of three articles about the different RAD approaches in Molecular
Ecology:
</p>
<ul class="org-ul">
<li>Andrews' original paper, <b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12686/abstract">Recent novel approaches for population genomics
data analysis</a></b>
<ul class="org-ul">
<li>Review about NGS and population genomics
</li>
<li>Section about RAD-seq (in which they mention the issue with PCR duplicates)
</li>
<li>Interesting section about genotype calling
</li>
<li>"Luikart quoted Steve O’Brien (St.  Petersburg State University) who
recently said that &gt;90% of the cost of WGS is the bioinformatics,
emphasizing the need for scripting and programming skills."
</li>
</ul>
</li>
<li>Puritz's response, <b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12965/abstract">Demystifying the RAD fad</a></b>
<ul class="org-ul">
<li>Disagree with the view that the original RAD protocol (Miller 2007,
Baird 2008) minimize the impact of PCR duplicates compared to other RAD
methods
</li>
<li>Present additional biases in RAD-seq that should be considered when
choosing a protocol
</li>
<li>Review pros and cons of the different RAD protocols
</li>
</ul>
</li>
<li>Andrews' reply, <b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12964/abstract">Trade-offs and utility of alternative RADseq methods: Reply
to Puritz <i>et al.</i></a></b>
<ul class="org-ul">
<li>Agree with Puritz that researchers must understand thoroughly the different
RAD protocols and analysis before choosing one
</li>
<li>Address some key points they find unclear or potentially misleading in
their evaluation of techniques
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-41">
<h2 id="sec-41">Heliconius Genome Consortium (2012) - Nature <b>(§)</b></h2>
<div class="outline-text-2" id="text-41">
<p>
<b><a href="http://www.nature.com/nature/journal/v487/n7405/full/nature11041.html">Butterfly genome reveals promiscuous exchange of mimicry adaptations among
 species</a></b>
</p>
<ul class="org-ul">
<li>Study hybridization and introgression in butterfly
</li>
<li>RAD tags sequencing for linkage mapping to assign and order genome
sequences onto the chromosomes
</li>
<li>RAD resequencing to build a robust phylogenetic tree
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-42">
<h2 id="sec-42">Keller (2013) - Molecular Ecology</h2>
<div class="outline-text-2" id="text-42">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12083/abstract">Population genomic signatures of divergent adaptation, gene flow and hybrid
 speciation in the rapid radiation of Lake Victoria cichlid fishes</a></b>
</p>
</div>
<div class="outline-3" id="outline-container-sec-42-1">
<h3 id="sec-42-1"><span class="todo TODO">TODO</span> read</h3>
</div>
</div>
<div class="outline-2" id="outline-container-sec-43">
<h2 id="sec-43">Parchman (2013) - Molecular Ecology</h2>
<div class="outline-text-2" id="text-43">
<p>
<b><a href="http://onlinelibrary.wiley.com/doi/10.1111/mec.12201/abstract">The genomic consequences of adaptive divergence and reproductive isolation
 between species of manakins</a></b>
</p>
</div>
<div class="outline-3" id="outline-container-sec-43-1">
<h3 id="sec-43-1"><span class="todo TODO">TODO</span> read</h3>
</div>
</div>
<div class="outline-2" id="outline-container-sec-44">
<h2 id="sec-44">Heffelfinger (2014) - BMC Genomics <b>(Rev)</b> <b>(§)</b></h2>
<div class="outline-text-2" id="text-44">
<p>
<b><a href="http://www.biomedcentral.com/1471-2164/15/979">Flexible and scalable genotyping-by-sequencing strategies for population
 studies</a></b>
</p>
<ul class="org-ul">
<li>Review methods of genome complexity reduction by restriction enzymes to
perform genotyping-by-sequencing
</li>
<li>Modify the protocol by using blunt-end enzymes so that Illumina adaptors can
be used directly, dual barcoding system for multiplexing and bead-based
library preparation for in-solution size selection without the need for
columns and gels
</li>
<li>"the sequenced portion of the genome was adaptable by selecting enzymes based
on motif length, complexity, and methylation sensitivity"
</li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-45">
<h2 id="sec-45">Wu (2010) - BMC Genomics</h2>
<div class="outline-text-2" id="text-45">
<p>
<b><a href="http://www.biomedcentral.com/1471-2164/11/469">SNP discovery by high-throughput sequencing in soybean</a></b>
</p>
<ul class="org-ul">
<li>Pooled DNA fragment reduced representation library
</li>
<li>Solexa short read sequencing to identify SNPs
</li>
</ul>
</div>
</div>

</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://orgmode.org/">Org mode</a>
          in <a href="http://www.gnu.org/software/emacs/">Emacs</a>
          and <a href="http://getpelican.com/">Pelican</a>, a static site
          generator which uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>