#+Title: Bibliography notes
#+Summary: bibliographyNotes
#+URL: bibliography-notes.html
#+Save_as: bibliography-notes.html
#+Sortorder: 020
#+Slug: bibliographyNotes

The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by [[http://genome.cshlp.org/content/17/2/240.long][Miller et al. (2007)]] from Eric
A. Johnson's group.

There are important differences between methods based on *restriction site
polymorphism* (like [[http://genome.cshlp.org/content/17/2/240.long][Miller et al. 2007]]) and methods that use restriction sites
as a way to *decrease genomic complexity* before sequencing.

Below are bibliography notes related to the use of RAD tags.

* Miller et al. (2007) - Genome Research ([[http://genome.cshlp.org/content/17/2/240.long][link]])

Title: /Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers/

- Abstract
  + Eric A. Johnson's group at the University of Oregon
  + Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
    *genome-wide representation of every site of a particular restriction enzyme
    by short DNA tags*"
  + Here, looking for polymorphism that disrupt restriction sites (like RFLP,
    AFLP) but for many more loci than the previous methods.
  + Not sequencing but use of a micro-array platform
  + "*RAD markers [...] allow high-throughput, high-resolution genotyping in
    both model and nonmodel systems*"
- Interest of RAD tags
  + Genetic mapping (recombination mapping, association mapping) needs
    "high-throughput, cost-effective methods to identify and genotype
    polymorphisms"
  + Previous methods:
    - allele-specific oligonucleotide microarrays (thousands of SNPs scored per
      hybridization)
    - direct hybridization of labelled genomic DNA to short oligonucleotide
      arrays
    - single feature polymorphisms (SFP) can be discovered and typed in a similar
      way
  + Those methods need costly preliminary steps (sequencing, oligonucleotide
    synthesis) and are not easily feasible for non-model organisms
- Principle
  + Very good illustration in [[http://genome.cshlp.org/content/17/2/240.long#F1][Figure 1]].
  + Digest - link tags - shear DNA - purify tags - release tags - type tags
- Demonstration
  + Mapping of a breakpoint on a /Drosophila/ chromosome
  + Identification of the major lateral plate locus in three-spine stickleback
- Discussion
  + Here RAD markers use the *restriction site as the polymorphic marker* (no
    sequencing)
  + Easy to produce microarray type of genotyping resource
  + Bulk genotyping is possible
  + If markers location is determined, genetic mapping is possible
# Comment to force pandoc to initiate a new section
* Miller et al. (2007) - Genome Biology ([[http://genomebiology.com/content/8/6/R105][link]])

Title: /RAD marker microarrays enable rapid mapping of zebrafish mutations/

- Abstract
  + Developed a RAD marker microarray resource for the zebrafish
  + Bulk segregant approach to map mutations
  + Assay for individual RAD markers in pooled populations

** TODO Read in more details

* Baird et al. (2008) - PLoS ONE ([[http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376][link]])

Title: /Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers/

