#+Title: Bibliography notes
#+HTML_HEADER: <META NAME="Summary" CONTENT="bibliographyNotes">
#+URL: bibliography-notes.html
#+Save_as: bibliography-notes.html
#+Sortorder: 020
#+Slug: bibliographyNotes
#+OPTIONS: num:nil html-postamble:nil

Below are bibliography notes related to the use of RAD tags - *(§)*, *(§§)* and
*(§§§)* denotes papers of particular interest, and *(Rev)* denotes
reviews. Please note that this bibliography is not exhaustive!

* Miller (2007) - Genome Research *(§§§)*

*[[http://genome.cshlp.org/content/17/2/240.long][Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers]]*

- Summary
  + Eric A. Johnson's group at the University of Oregon
  + Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
    *genome-wide representation of every site of a particular restriction enzyme
    by short DNA tags*"
  + Here, looking for polymorphism that disrupt restriction sites (like RFLP,
    AFLP) but for many more loci than the previous methods.
  + Not sequencing but use of a micro-array platform
  + "*RAD markers [...] allow high-throughput, high-resolution genotyping in
    both model and nonmodel systems*"
- Interest of RAD tags
  + Genetic mapping (recombination mapping, association mapping) needs
    "high-throughput, cost-effective methods to identify and genotype
    polymorphisms"
  + Previous methods:
    - allele-specific oligonucleotide microarrays (thousands of SNPs scored per
      hybridization)
    - direct hybridization of labelled genomic DNA to short oligonucleotide
      arrays
    - single feature polymorphisms (SFP) can be discovered and typed in a similar
      way
  + Those methods need costly preliminary steps (sequencing, oligonucleotide
    synthesis) and are not easily feasible for non-model organisms
- Principle
  + Very good illustration in [[http://genome.cshlp.org/content/17/2/240.long#F1][Figure 1]].
  + Digest - link tags - shear DNA - purify tags - release tags - type tags
- Demonstration
  + Mapping of a breakpoint on a /Drosophila/ chromosome
  + Identification of the major lateral plate locus in three-spine stickleback
- Discussion
  + Here RAD markers use the *restriction site as the polymorphic marker* (no
    sequencing)
  + Easy to produce microarray type of genotyping resource
  + Bulk genotyping is possible
  + If markers location is determined, genetic mapping is possible

* Miller (2007) - Genome Biology *(§)*

*[[http://genomebiology.com/content/8/6/R105][RAD marker microarrays enable rapid mapping of zebrafish mutations]]*

- Developed a RAD marker microarray resource for the zebrafish
- Bulk segregant approach to map mutations
- Assay for individual RAD markers in pooled populations

* Lewis (2007) - Genetics

*[[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/][High-Density Detection of Restriction-Site-Associated DNA Markers for Rapid
Mapping of Mutated Loci in Neurospora]]*

- Micro-array based RAD markers in /Neurospora crassa/

* Baird (2008) - PLoS One *(§§)*

*[[http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376][Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers]]*

- RAD tags associated with Illumina sequencing for SNP discovery
- SNP identified by sequenced polymorphism and disrupted restriction sites
- Marker density can be tuned by the choice of the enzymes (and their number)
- Samples can be multiplexed by barcoding
- Useful with or without a reference genome

* Hohenlohe (2010) - PLoS Genetics *(§§§)*

*[[http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862][Population Genomics of Parallel Adaptation in Threespine Stickleback using
Sequenced RAD Tags]]*

A major paper about sequenced RAD tags.

- Use natural populations, not lab crosses
- Genome scan of nucleotide diversity and differentiation in 5 natural
  populations of threespine stickleback
- Illumina-sequenced RAD tags
- Identify and type more than 45000 SNPs in each of 100 individuals
- Genomic regions with balancing and divergent selection, consistent across
  populations
- Identify the genes in regions of interest (regions of parallel
  differentiation across independent populations)
- Methods:
  - Genome is available
  - Barcoded sample
  - Kernel smoothing and permutation testing

* Hohenlohe (2011) - Molecular Ecology Resources *(§)*

*[[http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract][Next-generation RAD sequencing identifies thousands of SNPs for assessing
 hybridization between rainbow and westslope cutthroat trout]]*

- Nonmodel organism (no genome available)
- Salmonid fish, several whole-genome duplication
- RAD sequencing to identify SNP loci with fixed allelic differences between
  introduced rainbow trout and native westslope cutthroat trout
- Use 24 barcode-labelled individuals
- Use Stacks for analysis (genotyping)

* Emerson (2011) - PNAS *(§§)*

*[[http://www.pnas.org/content/107/37/16196.abstract][Resolving postglacial phylogeography using high-throughput sequencing]]*

- Fine-scale phylogenetic relationships between recently diverged (from a
  refugium after the recession of the Laurentide ice sheet 22000-19000 BP)
  populations of pitcher plant mosquito
- Discussed also in [[http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0034241][Puritz 2012]] ("Next-Generation Phylogeography: A Targeted
  Approach for Multilocus Sequencing of Non-Model Organisms") in which they use
  454 sequencing:
  #+BEGIN_QUOTE
  "However, few of [the NGS] technologies have been applied in phylogeographic
  studies, with the exception of Emerson et al. (42) that used
  restriction-site-associated DNA tags (RAD tags) (43,44) to determine the
  evolutionary relationship between recently diverged populations of pitcher
  plant mosquitos. This methodology can genotype multiple populations at
  thousands of SNP loci simultaneously, but have limited ability to survey
  large sample sizes within populations because of the cost. For example,
  Emerson et al. genotyped 21 different population at ,13,627 different SNPs
  but with only 6 individuals per population the authors were forced to
  restrict their analysis to 3,741 SNPs ‘‘that were fixed within at least two
  populations and were variable among populations’’ and generate one consensus
  sequence per locus per population (42). Our targeted 454 sequencing
  methodology offers a compromise with the ability to sequence a reasonable
  sample size (20 individuals) from one population for 10 different nDNA loci
  (,400 bp each) in 1/16th of a plate of 454 sequencing, at lower cost than a
  RAD tag analysis."
  #+END_QUOTE

* Etter (2011) - PLoS One

*[[http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561][Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads]]*

- Use paired-end sequencing of RAD fragments to build contigs on the side which
  is cut by random shearing
- Identify SNP and determine haplotype in threespine stickleback
- Produce overlapping contigs of several hundred nucleotides in /E. coli/ and
  threespine stickleback
- A circularization step can allow a local assembly of up to 5 kb

* Etter (2011) - Methods in Molecular Biology *(Rev)*

*[[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3658458/][SNP Discovery and Genotyping for Evolutionary Genetics Using RAD Sequencing]]*

- Review chapter about RAD genotyping (RAD with Illumina sequencing) or
  *RAD-seq*

* Davey (2011) - Briefings in Functional Genomics *(Rev)*

*[[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3080771/][RADSeq: next-generation population genetics]]*

- Review about RAD-seq and its use for population genomics
- Interest for non-model organisms, wild populations and ecological population
  genomics

* Davey (2011) - Nature Reviews Genetics *(Rev)* *(§)*

*[[http://www.nature.com/nrg/journal/v12/n7/full/nrg3012.html][Genome-wide genetic marker discovery and genotyping using
next-generation sequencing]]*

- Review about NGS methods that use restriction enzyme digestion to reduce the
  genome complexity
- Include reduced-representation sequencing using reduced-representation
  libraries (RRLs) or complexity reduction of polymorphic sequences (CRoPS)
- Include RAD-seq
- Include low coverage genotyping

* O'Rourke (2011) - Genetics

*[[http://www.ncbi.nlm.nih.gov/pubmed/21900274][Rapid Mapping and Identification of Mutations in /Caenorhabditis elegans/ by
Restriction Site-Associated DNA Mapping and Genomic Interval Pull-Down Sequencing]]*

- Forward genetic screens in /C. elegans/ (association between phenotypes and
  mutated genes)
- Use RAD-seq for rapid mapping of the mutations
- Use genomic interval pull-down sequencing (GIPS) to capture and sequence
  large portions of a mutant genome (ca. Mb)

* Pfender (2011) - Theor. Appl. Genet. *(§)*

*[[http://www.ncbi.nlm.nih.gov/pubmed/21344184][Mapping with RAD (restriction-site associated DNA) markers to rapidly identify
QTL for stem rust resistance in /Lolium perenne/]]*

- Use RAD-seq to produce maps and identify QTL (in association with simple
  sequence repeats (SSR) and sequence-tagged sites (STS))
- Reads from parental plants (2) and from F1 (188)
- High variability in the number of reads per F1 (14000 to 935000)

* Chong (2011) - Bioinformatics

*[[http://bioinformatics.oxfordjournals.org/content/28/21/2732.full][Rainbow: an integrated tool for efficient clustering and
assembling RAD-seq reads]]*

- Tools to assemble RAD reads

* Flight (2012) - Integrative and Comparative Biology *(§)*

*[[http://www.ncbi.nlm.nih.gov/pubmed/22767487][Genetic Variation in the Acorn Barnacle from Allozymes
to Population Genomics]]*

- Use allozymes, microsatellites, mtDNA and RADseq
- RADseq on pooled samples (20 individuals per pool) of 3 populations (2 US, 1
  UK)
- Use SOAPdenovo assembly
- Fst outlier analysis (using unbiased estimates and empirical distribution,
  with the 1% tail of the distribution being the outliers)

* Hohenlohe (2012) - Phil. Trans. R. Soc. B *(§§)*

*[[http://www.ncbi.nlm.nih.gov/pubmed/22201169][Extensive linkage disequilibrium and parallel adaptive divergence across
threespine stickleback genomes]]*

- Study patterns of local and long-distance linkage desequilibrium (LD) across
  oceanic and freshwater populations of threespine stickleback
- Look for association between LD and signatures of divergent selection
- Assess the role of recombination rate variation in generating LD patterns
- Reuse RAD-seq data from Hohenlohe 2010 (PLoS Genetics) and also generate new
  data with a modified RAD-seq protocol on 87 F2 individuals and the parents
  and F1 individuals
- Align raw reads to reference genome with Bowtie
- Use likelihood methods to infer the genotype at each nucleotide

* Houston (2012) - BMC Genomics *(§)*

*[[http://www.biomedcentral.com/1471-2164/13/244][Characterisation of QTL-linked and genome-wide restriction site-associated DNA
(RAD) markers in farmed Atlantic salmon]]*

- RAD-seq on two Atlantic salmon families
- Offspring were classified into resistant and susceptible to a disease, and
  for each family DNA from the parents and 7 offsprings of each phenotype was
  sequenced by RAD-seq in multiplexed pools
- Use of pedigreed samples allowed to distinguish segregating SNPs from
  putative paralogous sequences (recent genome duplication in salmonids)
- 50 SNPs linked to QTL and a subset used for high-throughput genotyping across
  large commercial populations of disease-challenged fry

* Peterson (2012) - PLoS One *(§§§)*

*[[http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0037135][Double Digest RADseq: An Inexpensive Method for /De Novo/ SNP Discovery and
Genotyping in Model and Non-Model Species]]*

- Introduce the double-digest method
- Combinatorial barcoding for larger number of individuals per multiplex
- Detailed pipeline of reads analysis
- Also compare the observed size distribution with /in silico/ digestion and
  statistical models of size selection

* Rubin (2012) - PLoS One *(§§)*

*[[http://journals.plos.org/plosone/article?id%3D10.1371/journal.pone.0033394][Inferring Phylogenies from RAD Sequence Data]]*

- Simulate RAD-seq data collection for interspecific phylogeny reconstruction
  in /Drosophila/, mammals and yeast
- Develop a workflow to test if informative data can be gained from RAD-seq and
  used for accurate reconstruction of "known" phylogenies
- RAD-seq seems promising for phylogenies in younger clades (e.g. 60 My vs 100
  My) in which enough orthologous restriction sites are retained across species

