#+Title: Bibliography notes
#+Summary: bibliographyNotes
#+URL: bibliography-notes.html
#+Save_as: bibliography-notes.html
#+Sortorder: 020
#+Slug: bibliographyNotes

The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by [[http://genome.cshlp.org/content/17/2/240.long][Miller et al. (2007)]] from Eric
A. Johnson's group.

There are important differences between methods based on *restriction site
polymorphism* (like [[http://genome.cshlp.org/content/17/2/240.long][Miller et al. 2007]]) and methods that use restriction sites
as a way to *decrease genomic complexity* before sequencing.

Below are bibliography notes related to the use of RAD tags.

* *Miller et al. (2007) - Genome Research* ([[http://genome.cshlp.org/content/17/2/240.long][link]])                     :major:

*Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers*

- Summary
  + Eric A. Johnson's group at the University of Oregon
  + Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
    *genome-wide representation of every site of a particular restriction enzyme
    by short DNA tags*"
  + Here, looking for polymorphism that disrupt restriction sites (like RFLP,
    AFLP) but for many more loci than the previous methods.
  + Not sequencing but use of a micro-array platform
  + "*RAD markers [...] allow high-throughput, high-resolution genotyping in
    both model and nonmodel systems*"
- Interest of RAD tags
  + Genetic mapping (recombination mapping, association mapping) needs
    "high-throughput, cost-effective methods to identify and genotype
    polymorphisms"
  + Previous methods:
    - allele-specific oligonucleotide microarrays (thousands of SNPs scored per
      hybridization)
    - direct hybridization of labelled genomic DNA to short oligonucleotide
      arrays
    - single feature polymorphisms (SFP) can be discovered and typed in a similar
      way
  + Those methods need costly preliminary steps (sequencing, oligonucleotide
    synthesis) and are not easily feasible for non-model organisms
- Principle
  + Very good illustration in [[http://genome.cshlp.org/content/17/2/240.long#F1][Figure 1]].
  + Digest - link tags - shear DNA - purify tags - release tags - type tags
- Demonstration
  + Mapping of a breakpoint on a /Drosophila/ chromosome
  + Identification of the major lateral plate locus in three-spine stickleback
- Discussion
  + Here RAD markers use the *restriction site as the polymorphic marker* (no
    sequencing)
  + Easy to produce microarray type of genotyping resource
  + Bulk genotyping is possible
  + If markers location is determined, genetic mapping is possible
# Comment to force pandoc to initiate a new section
* *Miller et al. (2007) - Genome Biology* ([[http://genomebiology.com/content/8/6/R105][link]])                      :major:

*RAD marker microarrays enable rapid mapping of zebrafish mutations*

- Developed a RAD marker microarray resource for the zebrafish
- Bulk segregant approach to map mutations
- Assay for individual RAD markers in pooled populations

** TODO Read in more details

* Lewis et al. (2007) - Genetics ([[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/][link]])

*High-Density Detection of Restriction-Site-Associated DNA Markers for Rapid
Mapping of Mutated Loci in Neurospora*

- Micro-array based RAD markers in /Neurospora crassa/


* *Baird et al. (2008) - PLoS ONE* ([[http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376][link]])                             :major:

*Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers*

- RAD tags associated with Illumina sequencing for SNP discovery
- SNP identified by sequenced polymorphism and disrupted restriction sites
- Marker density can be tuned by the choice of the enzymes (and their number)
- Samples can be multiplexed by barcoding
- Useful with or without a reference genome

* *Hohenlohe et al. (2010) - PLoS Genetics* ([[http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862][link]])                    :major:

*Population Genomics of Parallel Adaptation in Threespine Stickleback using
Sequenced RAD Tags*

A major paper about sequenced RAD tags.

- Use natural populations, not lab crosses
- Genome scan of nucleotide diversity and differentiation in 5 natural
  populations of threespine stickleback
- Illumina-sequenced RAD tags
- Identify and type more than 45000 SNPs in each of 100 individuals
- Genomic regions with balancing and divergent selection, consistent across
  populations
- Identify the genes in regions of interest (regions of parallel
  differentiation across independent populations)
- Methods:
  - Genome is available
  - Barcoded sample
  - Kernel smoothing and permutation testing

* *Hohenlohe et al. (2011) - Molecular Ecology Resources* ([[http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract][link]])      :major:

*Next-generation RAD sequencing identifies thousands of SNPs for assessing
 hybridization between rainbow and westslope cutthroat trout*

- Nonmodel organism (no genome available)
- Salmonid fish, several whole-genome duplication
- RAD sequencing to identify SNP loci with fixed allelic differences between
  introduced rainbow trout and native westslope cutthroat trout
- Use 24 barcode-labelled individuals
- Use Stacks for analysis (genotyping)

* Etter et al. (2011) - PLoS ONE ([[http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561][link]])

*Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads*

- Use paired-end sequencing of RAD fragments to build contigs on the side which
  is cut by random shearing
- Identify SNP and determine haplotype in threespine stickleback
- Produce overlapping contigs of several hundred nucleotides in /E. coli/ and
  threespine stickleback
- A circularization step can allow a local assembly of up to 5 kb

