<!DOCTYPE html>
<html lang="en">
  
  <head>
    <meta charset="utf-8" />
    <title>Draft notes</title>
    <link rel="stylesheet" href="./theme/css/main.css" />
    <!--[if IE]>
        <script src="http://html5shiv.googlecode.com/svn/trunk/html5.js"></script>
        <![endif]-->
  </head>

  <body id="index" class="home">
    <header id="banner" class="body">
      <h1><a href="./">Roscoff course 2015 - RAD tags </a></h1>
      <nav><ul>
<!-- http://stackoverflow.com/questions/18520046/how-can-i-control-the-order-of-pages-from-within-a-pelican-article-category -->
          <li><a href="./index.html">About</a></li>
          <li><a href="./schedule.html">Course schedule</a></li>
          <li><a href="./bibliography-notes.html">Bibliography notes</a></li>
      </ul></nav>
    </header><!-- /#banner -->

<section id="content" class="body">
    <!-- <h1 class="entry-title">Draft notes</h1> -->
    
    


<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#sec-1">1. Course about RAD-tag sequencing - Roscoff 2015</a>
<ul>
<li><a href="#sec-1-1">1.1. Problematic</a></li>
<li><a href="#sec-1-2">1.2. Presentation of the RAD-tag approach (history)</a></li>
<li><a href="#sec-1-3">1.3. Use of RAD tags</a></li>
<li><a href="#sec-1-4">1.4. Issues</a></li>
<li><a href="#sec-1-5">1.5. Technical points</a></li>
<li><a href="#sec-1-6">1.6. Pipeline</a></li>
<li><a href="#sec-1-7">1.7. Presentation</a></li>
<li><a href="#sec-1-8">1.8. Intregated studies</a></li>
</ul>
</li>
<li><a href="#sec-2">2. Practicals</a>
<ul>
<li><a href="#sec-2-1">2.1. Teaching objectives</a></li>
<li><a href="#sec-2-2">2.2. Available data</a></li>
<li><a href="#sec-2-3">2.3. Connection to ABiMS</a></li>
</ul>
</li>
<li><a href="#sec-3">3. Important people and groups</a>
<ul>
<li><a href="#sec-3-1">3.1. Three central groups</a></li>
<li><a href="#sec-3-2">3.2. People</a></li>
</ul>
</li>
<li><a href="#sec-4">4. Other interesting papers</a></li>
</ul>
</div>
</div>
<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1"><span class="section-number-2">1</span> Course about RAD-tag sequencing - Roscoff 2015</h2>
<div class="outline-text-2" id="text-1">
</div><div class="outline-3" id="outline-container-sec-1-1">
<h3 id="sec-1-1"><span class="section-number-3">1.1</span> Problematic</h3>
<div class="outline-text-3" id="text-1-1">
<ul class="org-ul">
<li>High coverage
</li>
<li>Genome reduction
</li>
<li>Non-model species
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-2">
<h3 id="sec-1-2"><span class="section-number-3">1.2</span> Presentation of the RAD-tag approach (history)</h3>
<div class="outline-text-3" id="text-1-2">
<ul class="org-ul">
<li>Protocol
</li>
<li>Single or double digest
</li>
<li>Single end or paired ends
</li>
<li>Advantage over SNP-chip: no prior knowledge of the genome is needed, less
costly to put in place for non-model species
</li>
<li>As explained in Peterson 2012, SNP-chip or similar, once designed, can only
test for the variability which was known at design time and cannot detect
rare alleles or population-specific variants, even though those can be of
high interest in population genetics studies
</li>
<li>Peterson 2012 introduces the double-digest method for reduced representation
but increased coverage
</li>
</ul>
<p>
The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by <a href="http://genome.cshlp.org/content/17/2/240.long">Miller (2007)</a> from Eric
A. Johnson's group.
</p>
<p>
There are important differences between methods based on <b>restriction site
polymorphism</b> (like <a href="http://genome.cshlp.org/content/17/2/240.long">Miller 2007</a>) and methods that use restriction sites as a
way to <b>decrease genomic complexity</b> before sequencing (RAD-seq). While
restriction site polymorphism was historically (in 2007!) used first, mainly
the RAD-seq method and its derivative are used at present.
</p>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-3">
<h3 id="sec-1-3"><span class="section-number-3">1.3</span> Use of RAD tags</h3>
<div class="outline-text-3" id="text-1-3">
<ul class="org-ul">
<li>SNP discovery
</li>
<li>Population genetics
</li>
<li>Functional genomics
</li>
<li>Phylogeography (Emerson 2011, discussed also in Puritz 2012)
</li>
<li>Phylogeny (Rubin 2012)
</li>
<li>historical population structure and recent colonization history (useful for
hydrothermal ecosystems?) (Catchen 2013 Mol Ecol)
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-4">
<h3 id="sec-1-4"><span class="section-number-3">1.4</span> Issues</h3>
<div class="outline-text-3" id="text-1-4">
<ul class="org-ul">
<li>Variation of coverage between libraries (e.g. Pfender 2011, Bruneaux 2013)
</li>
<li>The second read on paired reads is not at a constant location - ddRAD enables
to have stacks from both sides
</li>
<li>Pooled population samples: how many individuals to pool? How deep to
sequence? Find the paper I sent to Anti where there was ann excel sheet with
formulas to calculate the power of different sequencing designs
</li>
<li>Example of Bruneaux 2013 with low coverage: don't try to do too many
populations! Focus on simple design to get good coverage.
</li>
<li>DeFaveri 2013 paper: SNPs not as good as microsatellites for fine-scale
structure, can it be balanced by very large number of SNPs?
</li>
<li>Information content per SNP is lower that per microsats, but large numbers
can probably tip the balance towards high number of SNPs? Look for papers
about that? Shannon information theory?
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-5">
<h3 id="sec-1-5"><span class="section-number-3">1.5</span> Technical points</h3>
<div class="outline-text-3" id="text-1-5">
<ul class="org-ul">
<li>If reference genome, no need for assembly, raw reads can be mapped back to
the genome (Hohenlohe 2012)
</li>
<li>Likelihood methods can be used to determine the probability of each genotype
at each nucleotide position (e.g. Hohenlohe 2012 but also others I think)
</li>
<li>Paired-ends: use depends on whether it is RAD or ddRAD. With RAD, 2nd read
can be used for contigs. With ddRAD, 2nd read also provides stacks. The
paired information provides data about genomic distance (because of the
size-selection step in the sample preparation, cf. the observed size
distribution for the fragments observed for the 9spine
</li>
<li>Gautier 2013 paper about allele frequency estimate accuracy (pool vs
individual), application to calculate accuracy
</li>
<li>Davey 2013: issues with RAD genotyping
</li>
<li>Use of black boxes (e.g. stacks): pros and cons
<ul class="org-ul">
<li>pros: the people who made the software knew what they do, enable to perform
rigorous statistical analyses, …
</li>
<li>cons: less control/understanding from the user, if bad choice of parameters
can be not applicable to one's dataset
</li>
<li>the user should understand what is done!
</li>
<li>in the end, we always rely on a black box but at a different level
(e.g. STACKS, blast, R, python, linux kernel, …)
</li>
<li>find a balance: if we trust the writers of the black box, we must also be
confident that we use it properly
</li>
</ul>
</li>
<li>Genotype calling: likelihood-based and Bayesian approaches, quick review in
Andrews 2014 "Recent novel approaches for population genomics data analysis"
</li>
<li>Puritz 2014 (reply to Andrews): pros and cons of several RAD protocols,
useful to choose a protocol for a given question
</li>
<li>Pukk 2015: RAD study doesn't have to be super fancy to give interesting
results, e.g. with ion torrent and not very high coverage robust diagnostic
SNPs between populations can still be detected
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-6">
<h3 id="sec-1-6"><span class="section-number-3">1.6</span> Pipeline</h3>
<div class="outline-text-3" id="text-1-6">
<ul class="org-ul">
<li>Cleaning
</li>
<li>Demultiplex and remove adapters
</li>
<li>Detailed pipeline for ddRAD in Peterson 2012, with very interesting choices
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-7">
<h3 id="sec-1-7"><span class="section-number-3">1.7</span> Presentation</h3>
<div class="outline-text-3" id="text-1-7">
<ul class="org-ul">
<li>Introduce the 9spine system, gigantism, different behaviour, plate morphs,
short reference to threespine also, marine vs freshwater (specify Baltic Sea
salinity is low)
</li>
<li>Talk about the sample preparation (by Anti, cf. his ppt presentation)
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-8">
<h3 id="sec-1-8"><span class="section-number-3">1.8</span> Intregated studies</h3>
<div class="outline-text-3" id="text-1-8">
<ul class="org-ul">
<li>Bruneaux 2013
</li>
<li>Hess 2013 (sea lamprey)
</li>
<li>Others…
</li>
</ul>
</div>
</div>
</div>
<div class="outline-2" id="outline-container-sec-2">
<h2 id="sec-2"><span class="section-number-2">2</span> Practicals</h2>
<div class="outline-text-2" id="text-2">
<p>
Pipelines include STACKS and dDocent
<a href="https://ddocent.wordpress.com/ddocent-pipeline-user-guide/">https://ddocent.wordpress.com/ddocent-pipeline-user-guide/</a>.
</p>
</div>
<div class="outline-3" id="outline-container-sec-2-1">
<h3 id="sec-2-1"><span class="section-number-3">2.1</span> Teaching objectives</h3>
<div class="outline-text-3" id="text-2-1">
<ul class="org-ul">
<li>Sequence quality check: trimming, quality control
</li>
<li>Stack building (STACKS or other tools, with or without paired-ends
information)
</li>
<li>SNP calling and genotyping (VCFtools)
</li>
<li>Allele frequency estimates (individuals or pools?)
</li>
<li>Population genetics: comparison of microsats and SNPs trees (cf. Anti)
</li>
<li>Population genomics: localisation, genome profiling and Gst calculation,
kernel smoothing
</li>
<li>Functional analysis (outside the scope of RAD-tags <i>sensus stricto</i>)
</li>
<li>Predicting the number of fragmnts based on published genome sequence and
restriction sites with python scripts
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-2-2">
<h3 id="sec-2-2"><span class="section-number-3">2.2</span> Available data</h3>
<div class="outline-text-3" id="text-2-2">
<ul class="org-ul">
<li>Baird 2008: short reads for three-spines and <i>Neurospora crassa</i>
</li>
<li>Hohenlohe 2010: short reads for three-spines (population genomics of parallel
evolution)
</li>
<li>Hohenlohe 2011: short reads for rainbow trout and westslope cutthroat trout
(SNP identification for hybridization diagnostic)
</li>
<li>Hess 2013: sea lamprey
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-2-3">
<h3 id="sec-2-3"><span class="section-number-3">2.3</span> Connection to ABiMS</h3>
<div class="outline-text-3" id="text-2-3">
<ul class="org-ul">
<li><a href="http://abims.sb-roscoff.fr/resources/cluster/howto">Cluster HowTo</a>
</li>
<li><a href="http://application.sb-roscoff.fr/ganglia/">Cluster load monitoring</a>
</li>
</ul>
</div>
</div>
</div>
<div class="outline-2" id="outline-container-sec-3">
<h2 id="sec-3"><span class="section-number-2">3</span> Important people and groups</h2>
<div class="outline-text-2" id="text-3">
</div><div class="outline-3" id="outline-container-sec-3-1">
<h3 id="sec-3-1"><span class="section-number-3">3.1</span> Three central groups</h3>
<div class="outline-text-3" id="text-3-1">
<p>
To the best of my knowledge, there are three groups which are involved in the
origin of the RAD tags method and which have produced a lot of the initial
papers. They are also producing a lot of the software that can be used for RAD
data analysis. The groups are:
</p>
<ul class="org-ul">
<li>Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon): <a href="http://molbio.uoregon.edu/johnson/">link</a>
</li>
<li>William A. Cresko's group (Institute of Ecology and Evolution, U. Oregon):
<a href="http://creskolab.uoregon.edu/">link</a>. The Cresko lab is responsible for the <a href="http://creskolab.uoregon.edu/stacks/">Stacks</a> pipeline that can be used
to analyse RAD data.
</li>
<li>Paul A. Hohenlohe (Depts of Biological Science and Statistics, U. Idaho):
<a href="http://webpages.uidaho.edu/hohenlohe/index.html">link</a>. They also have some software and their news page has some interesting
information about what's going on with RAD.
</li>
</ul>
</div>
</div>
<div class="outline-3" id="outline-container-sec-3-2">
<h3 id="sec-3-2"><span class="section-number-3">3.2</span> People</h3>
<div class="outline-text-3" id="text-3-2">
<ul class="org-ul">
<li>Miller Michael R.
</li>
<li>Johnson Eric A.
</li>
<li>Cresko William A.
</li>
<li>Baird Nathan A.
</li>
<li>Catchen Julian M.
</li>
</ul>
</div>
</div>
</div>
<div class="outline-2" id="outline-container-sec-4">
<h2 id="sec-4"><span class="section-number-2">4</span> Other interesting papers</h2>
<div class="outline-text-2" id="text-4">
<ul class="org-ul">
<li>Hohenlohe 2010, "Using population genomics to detect selection in natural
populations: key concepts and methodological considerations" (<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/">link</a>).
</li>
<li>Ellengren 2014, "Genome sequencing and population genomics in non-model
organisms" (<a href="http://www.sciencedirect.com/science/article/pii/S0169534713002310">link</a>)
</li>
</ul>
</div>
</div>

</section>

    <!-- <section id="extras" class="body"> -->
    <!--  -->
    <!--  -->
    <!-- </section><\!-- /#extras -\-> -->
    
    <footer id="contentinfo" class="body">
      <address id="about" class="vcard body">
	<p>
          Generated with <a href="http://getpelican.com/">Pelican</a>, a
          static site generator which
          uses <a href="http://python.org">Python</a>.
	</p>
        <p>
	  Theme modified from the original <i>notmyidea</i> theme
	  by <a href="http://coding.smashingmagazine.com/2009/08/04/designing-a-html-5-layout-from-scratch/">Smashing
	  Magazine</a>.
	</p>
      </address><!-- /#about -->
    </footer><!-- /#contentinfo -->

    <!--  -->
    <!--  -->
    <!--  -->

  </body>
</html>