#+Title: Draft notes
#+Summary: draftNotes
#+URL: draft-notes.html
#+Save_as: draft-notes.html
#+Sortorder: 030
#+Slug: draftNotes
#+OPTIONS: html-postamble:nil

* Course about RAD-tag sequencing - Roscoff 2015

** Problematic
- High coverage
- Genome reduction
- Non-model species

** Presentation of the RAD-tag approach (history)
- Protocol
- Single or double digest
- Single end or paired ends
- Advantage over SNP-chip: no prior knowledge of the genome is needed, less
  costly to put in place for non-model species
- As explained in Peterson 2012, SNP-chip or similar, once designed, can only
  test for the variability which was known at design time and cannot detect
  rare alleles or population-specific variants, even though those can be of
  high interest in population genetics studies
- Peterson 2012 introduces the double-digest method for reduced representation
  but increased coverage
** Use of RAD tags
- SNP discovery
- Population genetics
- Functional genomics
- Phylogeography (Emerson 2011, discussed also in Puritz 2012)
- Phylogeny (Rubin 2012)
** Issues
- Variation of coverage between libraries (e.g. Pfender 2011, Bruneaux 2013)
- The second read on paired reads is not at a constant location - ddRAD enables
  to have stacks from both sides
- Pooled population samples: how many individuals to pool? How deep to
  sequence? Find the paper I sent to Anti where there was ann excel sheet with
  formulas to calculate the power of different sequencing designs
- Example of Bruneaux 2013 with low coverage: don't try to do too many
  populations! Focus on simple design to get good coverage.
** Technical points
- If reference genome, no need for assembly, raw reads can be mapped back to
  the genome (Hohenlohe 2012)
- Likelihood methods can be used to determine the probability of each genotype
  at each nucleotide position (e.g. Hohenlohe 2012 but also others I think)
- Paired-ends: use depends on whether it is RAD or ddRAD. With RAD, 2nd read
  can be used for contigs. With ddRAD, 2nd read also provides stacks. The
  paired information provides data about genomic distance (because of the
  size-selection step in the sample preparation, cf. the observed size
  distribution for the fragments observed for the 9spine
** Pipeline
- Cleaning
- Demultiplex and remove adapters
- Detailed pipeline for ddRAD in Peterson 2012, with very interesting choices
** Presentation
- Introduce the 9spine system, gigantism, different behaviour, plate morphs,
  short reference to threespine also, marine vs freshwater (specify Baltic Sea
  salinity is low)
- Talk about the sample preparation (by Anti, cf. his ppt presentation)
* Practicals

** Teaching objectives
- Sequence quality check: trimming, quality control
- Stack building (STACKS or other tools, with or without paired-ends
  information)
- SNP calling and genotyping (VCFtools)
- Allele frequency estimates (individuals or pools?)
- Population genetics: comparison of microsats and SNPs trees (cf. Anti)
- Population genomics: localisation, genome profiling and Gst calculation,
  kernel smoothing
- Functional analysis (outside the scope of RAD-tags /sensus stricto/)
- Predicting the number of fragmnts based on published genome sequence and
  restriction sites with python scripts

** Available data
- Baird 2008: short reads for three-spines and /Neurospora crassa/
- Hohenlohe 2010: short reads for three-spines (population genomics of parallel
  evolution)
- Hohenlohe 2011: short reads for rainbow trout and westslope cutthroat trout
  (SNP identification for hybridization diagnostic)

** Connection to ABiMS
- [[http://abims.sb-roscoff.fr/resources/cluster/howto][Cluster HowTo]]
- [[http://application.sb-roscoff.fr/ganglia/][Cluster load monitoring]]


* Important people and groups

** Three central groups

To the best of my knowledge, there are three groups which are involved in the
origin of the RAD tags method and which have produced a lot of the initial
papers. They are also producing a lot of the software that can be used for RAD
data analysis. The groups are:
- Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon): [[http://molbio.uoregon.edu/johnson/][link]]
- William A. Cresko's group (Institute of Ecology and Evolution, U. Oregon):
  [[http://creskolab.uoregon.edu/][link]]. The Cresko lab is responsible for the [[http://creskolab.uoregon.edu/stacks/][Stacks]] pipeline that can be used
  to analyse RAD data.
- Paul A. Hohenlohe (Depts of Biological Science and Statistics, U. Idaho):
  [[http://webpages.uidaho.edu/hohenlohe/index.html][link]]. They also have some software and their news page has some interesting
  information about what's going on with RAD.

** People
- Miller Michael R.
- Johnson Eric A.
- Cresko William A.
- Baird Nathan A.
- Catchen Julian M.

* Other interesting papers

- Hohenlohe 2010, "Using population genomics to detect selection in natural
  populations: key concepts and methodological considerations" ([[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/][link]]).
  An interesting review about population genomics.
