* Course about RAD-tag sequencing - Roscoff 2015

** Problematic
- High coverage
- Genome reduction
- Non-model species

** Presentation of the RAD-tag approach (history)
- Protocol
- Single or double digest
- Single end or paired ends

** Use of RAD tags
- SNP discovery
- Population genetics
- Functional genomics

* Practicals

** Teaching objectives
- Sequence quality check: trimming, quality control
- Stack building (STACKS or other tools, with or without paired-ends
  information)
- SNP calling and genotyping (VCFtools)
- Allele frequency estimates (individuals or pools?)
- Population genetics: comparison of microsats and SNPs trees (cf. Anti)
- Population genomics: localisation, genome profiling and Gst calculation,
  kernel smoothing
- Functional analysis (outside the scope of RAD-tags /sensus stricto/)
- Predicting the number of fragmnts based on published genome sequence and
  restriction sites with python scripts

* Bibliography

Important difference between methods based on restriction site polymorphism
(like Miller 2007 Genome Research) and methods that use restriction sites as a
way to decrease genomic complexity before seauencing.

** Miller 2007 Genome Research
*** Abstract
- Eric A. Johnson's group at the University of Oregon
- Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
  genome-wide representation of every site of a particular restriction enzyme
  by short DNA tags"
- Here, looking for polymorphism that disrupt restriction sites (like RFLP,
  AFLP) but for many more loci than the previous methods.
- Not sequencing but use of a micro-array platform
- "RAD markers [...] allow high-throughput, high-resolution genotyping in both
  model and nonmodel systems"
*** Interest of RAD tags
- Genetic mapping (recombination mapping, association mapping) needs
  "high-throughput, cost-effective methods to identify and genotype
  polymorphisms"
- Previous methods:
  + allele-specific oligonucleotide microarrays (thousands of SNPs scored per
    hybridization)
  + direct hybridization of labelled genomic DNA to short oligonucleotide
    arrays
  + single feature polymorphisms (SFP) can be discovered and typed in a similar
    way
- Those methods need costly preliminary steps (sequencing, oligonucleotide
  synthesis) and are not easily feasible for non-model organisms
*** Principle
- Very good illustration in Figure 1
*** TODO finish reading
