#+Title: Draft notes
#+Summary: draftNotes
#+URL: draft-notes.html
#+Save_as: draft-notes.html
#+Sortorder: 030
#+Slug: draftNotes
#+Status: hidden
#+OPTIONS: html-postamble:nil

* Course about RAD-tag sequencing - Roscoff 2015

** Problematic
- High coverage
- Genome reduction
- Non-model species

** Presentation of the RAD-tag approach (history)
- Protocol
- Single or double digest
- Single end or paired ends
- Advantage over SNP-chip: no prior knowledge of the genome is needed, less
  costly to put in place for non-model species
- As explained in Peterson 2012, SNP-chip or similar, once designed, can only
  test for the variability which was known at design time and cannot detect
  rare alleles or population-specific variants, even though those can be of
  high interest in population genetics studies
- Peterson 2012 introduces the double-digest method for reduced representation
  but increased coverage

The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by [[http://genome.cshlp.org/content/17/2/240.long][Miller (2007)]] from Eric
A. Johnson's group.

There are important differences between methods based on *restriction site
polymorphism* (like [[http://genome.cshlp.org/content/17/2/240.long][Miller 2007]]) and methods that use restriction sites as a
way to *decrease genomic complexity* before sequencing (RAD-seq). While
restriction site polymorphism was historically (in 2007!) used first, mainly
the RAD-seq method and its derivative are used at present.

** Use of RAD tags
- SNP discovery
- Population genetics
- Functional genomics
- Phylogeography (Emerson 2011, discussed also in Puritz 2012)
- Phylogeny (Rubin 2012)
- historical population structure and recent colonization history (useful for
  hydrothermal ecosystems?) (Catchen 2013 Mol Ecol)
** Issues
- Variation of coverage between libraries (e.g. Pfender 2011, Bruneaux 2013)
- The second read on paired reads is not at a constant location - ddRAD enables
  to have stacks from both sides
- Pooled population samples: how many individuals to pool? How deep to
  sequence? Find the paper I sent to Anti where there was ann excel sheet with
  formulas to calculate the power of different sequencing designs
- Example of Bruneaux 2013 with low coverage: don't try to do too many
  populations! Focus on simple design to get good coverage.
- DeFaveri 2013 paper: SNPs not as good as microsatellites for fine-scale
  structure, can it be balanced by very large number of SNPs?
- Information content per SNP is lower that per microsats, but large numbers
  can probably tip the balance towards high number of SNPs? Look for papers
  about that? Shannon information theory?
** Technical points
- If reference genome, no need for assembly, raw reads can be mapped back to
  the genome (Hohenlohe 2012)
- Likelihood methods can be used to determine the probability of each genotype
  at each nucleotide position (e.g. Hohenlohe 2012 but also others I think)
- Paired-ends: use depends on whether it is RAD or ddRAD. With RAD, 2nd read
  can be used for contigs. With ddRAD, 2nd read also provides stacks. The
  paired information provides data about genomic distance (because of the
  size-selection step in the sample preparation, cf. the observed size
  distribution for the fragments observed for the 9spine
- Gautier 2013 paper about allele frequency estimate accuracy (pool vs
  individual), application to calculate accuracy
- Davey 2013: issues with RAD genotyping
- Use of black boxes (e.g. stacks): pros and cons
  - pros: the people who made the software knew what they do, enable to perform
    rigorous statistical analyses, ...
  - cons: less control/understanding from the user, if bad choice of parameters
    can be not applicable to one's dataset
  - the user should understand what is done!
  - in the end, we always rely on a black box but at a different level
    (e.g. STACKS, blast, R, python, linux kernel, ...)
  - find a balance: if we trust the writers of the black box, we must also be
    confident that we use it properly
- Genotype calling: likelihood-based and Bayesian approaches, quick review in
  Andrews 2014 "Recent novel approaches for population genomics data analysis"
- Puritz 2014 (reply to Andrews): pros and cons of several RAD protocols,
  useful to choose a protocol for a given question
- Pukk 2015: RAD study doesn't have to be super fancy to give interesting
  results, e.g. with ion torrent and not very high coverage robust diagnostic
  SNPs between populations can still be detected
** Pipeline
- Cleaning
- Demultiplex and remove adapters
- Detailed pipeline for ddRAD in Peterson 2012, with very interesting choices
** Presentation
- Introduce the 9spine system, gigantism, different behaviour, plate morphs,
  short reference to threespine also, marine vs freshwater (specify Baltic Sea
  salinity is low)
- Talk about the sample preparation (by Anti, cf. his ppt presentation)
** Intregated studies
- Bruneaux 2013
- Hess 2013 (sea lamprey)
- Others...

* Practicals

Pipelines include STACKS and dDocent
https://ddocent.wordpress.com/ddocent-pipeline-user-guide/.

** Teaching objectives
- Sequence quality check: trimming, quality control
- Stack building (STACKS or other tools, with or without paired-ends
  information)
- SNP calling and genotyping (VCFtools)
- Allele frequency estimates (individuals or pools?)
- Population genetics: comparison of microsats and SNPs trees (cf. Anti)
- Population genomics: localisation, genome profiling and Gst calculation,
  kernel smoothing
- Functional analysis (outside the scope of RAD-tags /sensus stricto/)
- Predicting the number of fragmnts based on published genome sequence and
  restriction sites with python scripts

** Available data
- Baird 2008: short reads for three-spines and /Neurospora crassa/
- Hohenlohe 2010: short reads for three-spines (population genomics of parallel
  evolution)
- Hohenlohe 2011: short reads for rainbow trout and westslope cutthroat trout
  (SNP identification for hybridization diagnostic)
- Hess 2013: sea lamprey

** Connection to ABiMS
- [[http://abims.sb-roscoff.fr/resources/cluster/howto][Cluster HowTo]]
- [[http://application.sb-roscoff.fr/ganglia/][Cluster load monitoring]]

** Softwares needed

| Name                  | From | Download Data    | Source file           | On ABiMS                                                   | Role                      |
|-----------------------+------+------------------+-----------------------+------------------------------------------------------------+---------------------------|
| CAP3                  | [[http://seq.cs.iastate.edu/cap3.html][link]] | [2015-03-05 Thu] | cap3.linux.x86_64.tar | manual install (older version already installed)           | (raw) reads assembly      |
| STACKS                | [[http://creskolab.uoregon.edu/stacks/][link]] | [2015-03-05 Thu] | stacks-1.27.tar.gz    | manual install                                             | Complete RAD pipeline     |
| SciRoKo               | [[http://kofler.or.at/bioinformatics/SciRoKo/Download.html][link]] | -                | -                     | no (but we don't want, as it needs Mono)                   | Detect repeats            |
| Sputnik               | -    | -                | -                     | already installed                                          | Detect repeats            |
| Blast                 | -    | -                | -                     | already installed                                          | Mapping                   |
| fastq_quality_trimmer | -    | -                | -                     | already installed (but is it the same as the Galaxy tool?) | Quality trimming of reads |
| Maq v0.7.1            | -    | -                | -                     | already installed                                          | Variant calling           |
  
** Search for microsatellites

Kofler 2007 ([[http://bioinformatics.oxfordjournals.org/content/23/13/1683.full#T1][paper]]) - SciRoKo software But SciRoKo needs Mono to run on Linux.
Some alternatives are TRF (but its license prohibits de-compilation or
modification) and modified Sputnik II.

** Data used for the practicals

Raw reads files were generated by concatenating the files in
=~/documents/turku_lab/papers/completed/2012-03_ninespine_RAD_tags/dryad_submission/=
in the zip file =Raw Reads for Bruneaux et al. 2012.zip= by lane and read
direction. These files should be similar to the ones available on Dryad.

I generated the modified =s_5_2_sequence.fastq= by using a python script
(=generate_barcode_restriction_site_noise.py=) which would randomly replace the
first 6 nucleotide of each sequence by one of "A", "T", "G", "C", "" with a
probability of 0.001 (output to =s_5_2_sequence.lowQual.fastq=).

* Important people and groups

** Three central groups

To the best of my knowledge, there are three groups which are involved in the
origin of the RAD tags method and which have produced a lot of the initial
papers. They are also producing a lot of the software that can be used for RAD
data analysis. The groups are:
- Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon): [[http://molbio.uoregon.edu/johnson/][link]]
- William A. Cresko's group (Institute of Ecology and Evolution, U. Oregon):
  [[http://creskolab.uoregon.edu/][link]]. The Cresko lab is responsible for the [[http://creskolab.uoregon.edu/stacks/][Stacks]] pipeline that can be used
  to analyse RAD data.
- Paul A. Hohenlohe (Depts of Biological Science and Statistics, U. Idaho):
  [[http://webpages.uidaho.edu/hohenlohe/index.html][link]]. They also have some software and their news page has some interesting
  information about what's going on with RAD.

** People
- Miller Michael R.
- Johnson Eric A.
- Cresko William A.
- Baird Nathan A.
- Catchen Julian M.

* Other interesting papers

- Hohenlohe 2010, "Using population genomics to detect selection in natural
  populations: key concepts and methodological considerations" ([[http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/][link]]).
- Ellengren 2014, "Genome sequencing and population genomics in non-model
  organisms" ([[http://www.sciencedirect.com/science/article/pii/S0169534713002310][link]])

