#+Title: RAD-seq in Roscoff
#+Author: Matthieu Bruneaux
#+Date: 2015-03-10

* General introduction to the workshop

** Mini-workshop about ddRAD

*** Introduction about RAD-seq                                    :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- RAD? RAD-seq? ddRAD?
- Applications
- Workflow

*** Practicals                                                    :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- One complete project, from raw reads to final results
- Cherry-picking of some analysis steps
- Open questions

*** Objectives                                                    :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Overview of RAD-seq
- Arouse curiosity
- Give useful pointers

** Disclaimer about the speaker!

***                                                                 :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Not a population geneticist, not a bioinformatician
- Evolutionary biologist who dropped into a RAD-seq project when he was
  a small post-doc
- Some things said here are probably incorrect or plainly wrong!

* Introduction: RAD-seq and population genomics

** What are RAD markers?

*** Miller et al. 2007                                            :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/miller-2007a-title.png]]
#+END_CENTER

*** Description of RAD markers
- *Restriction site associated DNA* fragments
- Used with micro-array systems
- Similar to RFLP or AFLP, but many more markers

** RAD - Miller et al. 2007 (6 steps)

*** Digest - tag - shear                                    :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.80
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/miller-2007a-fig1a.png]]
#+END_CENTER

** RAD - Miller et al. 2007 (6 steps)

*** Purify - release - type                                 :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.75
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/miller-2007a-fig1b.png]]
#+END_CENTER

** RAD - Miller et al. 2007 (method summary)

*** Digest - tag - shear                                    :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.47
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/miller-2007a-fig1a.png]]
#+END_CENTER

*** Purify - release - type                                 :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.47
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/miller-2007a-fig1b.png]]
#+END_CENTER

*** Demonstration                                                 :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Mapping breakpoint on a /Drosophila/ chromosome
- Identification of the lateral plate locus in threespine stickleback

** RAD - Miller et al. 2007

*** Advantage of the method                                       :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Easy-to-produce genotyping resource for *non-model species*
- *Moderate cost*
- *Genetic mapping* possible (if markers location known)
- *Bulk genotyping* possible

*** But note that...                                              :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- At this point *the restriction site is the polymorphic marker*
- *One restriction enzyme* only is used

** What is RAD-seq?

*** Baird et al. 2008                                             :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-title.png]]
#+END_CENTER

*** RAD-seq                                                       :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- RAD fragments with *high-throughput sequencing* (Illumina)
- SNP identified by *sequence polymorphism* and *site disruption*
- Can be used *with or without reference genome*

** RAD-seq - Baird 2008

*** Digest - tag                                                   :B_column:
:PROPERTIES:
:BEAMER_env: column
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig1a.png]]
#+END_CENTER

** RAD-seq - Baird 2008

*** Pool - shear                                                 :B_column:
:PROPERTIES:
:BEAMER_env: column
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig1b.png]]
#+END_CENTER

** RAD-seq - Baird 2008

*** Second adapter                                               :B_column:
:PROPERTIES:
:BEAMER_env: column
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig1c.png]]
#+END_CENTER

** RAD-seq - Baird 2008

*** Amplification and sequencing                                   :B_column:
:PROPERTIES:
:BEAMER_env: column
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig1d.png]]
#+END_CENTER

** RAD-seq - Baird 2008

*** Method summary                                               :B_column:
:PROPERTIES:
:BEAMER_env: column
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig1.png]]
#+END_CENTER

** RAD-seq - Baird 2008

*** Demonstration                                           :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.4
:END:
- Discover *13000 SNP* in threespine stickleback and in /Neurospora/
- *Barcoding* system for multiplexing
- *Marker density* can be tuned by the choice of restriction enzyme

*** Threespine stickleback                                  :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.55
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig2.png]]
#+END_CENTER

** Population genomics of parallel adaptation - Hohenlohe 2010

*** A major paper                                                  :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/hohenlohe-2010-title.png]]
#+END_CENTER

*** Method                                                         :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Model: threespine stickleback
- Comparison of 3 freshwater and 2 marine populations
- 20 individuals per population, individual barcodes
- Single reads (not paired ends)

** Population genomics of parallel adaptation - Hohenlohe 2010

*** /Gasterosteus aculeatus/           :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.45
:END:
#+BEGIN_CENTER
[[file:images/baird-2008-fig2.png]]
#+END_CENTER

*** Locations                                                 :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.45
:END:
#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig1.png]]
#+END_CENTER

** Hohenlohe 2010

#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig2a.png]]
#+END_CENTER

** Hohenlohe 2010

#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig2b.png]]
#+END_CENTER

** Hohenlohe 2010 - Genome profiles

#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig3.png]]
#+END_CENTER

***                                                :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- A: number of RAD tags per 1Mb
- B: Coverage per RAD per individual in one run (16 individuals - black line is
  average)

** Hohenlohe 2010

***                                                         :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.55
:END:

#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig5.png]]
#+END_CENTER

*** Evidence for balancing selection                        :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.40
:END:
- A: Nucleotide diversity, B: heterozygosity across all five populations
  (blue), three FW (red) or two SW (green)
- C: Fst between FW and SW (blue), among FW (red) and among SW (green)
- Horizontal bars shows regions of significantly elevated or reduced values on
  the profile

** Hohenlohe 2010

*** Genome-wide differentiation among populations           :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.45
:END:
#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig6.png]]
#+END_CENTER

*** Differentiation among SW and FW, zoom on LG               :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.45
:END:
#+BEGIN_CENTER
[[file:images/hohenlohe-2010-fig8.png]]
#+END_CENTER

** Hohenlohe 2010

*** Highlights                                                    :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- RAD-seq on *natural populations*, 45000 SNPs in 100 individuals
- *Barcoded* samples
- Genome profiling, *kernel smoothing* and *permutation testing*

*** But note that...                                              :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Genome available
- Single reads

** What is paired-end RAD-seq?

*** Etter 2011                                                    :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/etter-2011-title.png]]
#+END_CENTER

*** Method                                                        :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Paired-end sequencing of RAD fragments to build *contigs* on the randomly
  sheared side
- Demonstration with threespine and /E. coli/ sequencing
- Up to 5kb contigs with circularization step

** Single-reads RAD-seq

#+BEGIN_CENTER
[[file:images/baird-2008-fig1d.png]]
#+END_CENTER

** Paired-ends RAD-seq

#+BEGIN_CENTER
[[file:images/etter-2011-fig1.png]]
#+END_CENTER

*** Notes                                                         :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- The *stacked end* is useful for *high coverage* work (SNP calling, allele
  frequency estimates)
- The *echelon end* is useful for contig building, but *base coverage is lower*

** What is double-digest RAD-seq?

*** Peterson et al. 2012                                          :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/peterson-2012-title.png]]
#+END_CENTER

*** Method
- Two enzyme double digest followed by precise size selection
- Library contains only fragments close to target size
- Read counts across regions are expected to be correlated between individuals

** Peterson 2012

*** Double digest RAD tag                                         :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/peterson-2012-fig1.png]]
#+END_CENTER

** What is paired-end double RAD?

*** Bruneaux et al. 2013                                          :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/bruneaux-2013-title.png]]
#+END_CENTER

*** Method                                                        :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Two enzyme double digestion
- Paired-end sequencing after size-selection
- *You will hear more about it soon* (see practicals)

** Uses of RAD tags

*** From Peterson 2012                                            :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/peterson-2012-fig2.png]]
#+END_CENTER

** There are also some potential issues...

*** Crucial to understand the potential biases of RAD tags
- *PCR-duplicates*
- *Individual vs pool genotyping* for allele frequencies
- Comparison *SNP vs microsat*

*** Needs for (bio)informatic analyses
- *Specific pipelines* have been developed (STACKS, Rainbow, dDocent)
- Usual NGS tools can be used
- Again, the most important is to *understand what is going on*

** Conclusion

*** In a nutshell                                                 :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- *RAD tags*: versatile method of *genome complexity reduction*
- *RAD-seq*: large scale discovery of SNPs, affordable
- Useful for both *model* and *non-model* organisms
- *Just a tool*: the downstream analyses are still *your expertise*

* Practicals

** Before starting the practicals

***                                                               :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
Any questions ?

** Practical plan

*** Complete analysis, from raw reads to results                  :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Reproduce results from Bruneaux et al. 2013
- From raw reads to final results
- Skipping some steps

*** Cherry picking some other analyses?                           :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- If we have time
- You can tell me what you would be interested in

** General workflow (1/2)

*** RAD-seq experiment                                            :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
1. DNA extraction (pooling?)
2. Digestion and adapter ligation (simple or double RAD? Barcodes?)
3. Size selection
4. Sequencing (single reads? double reads?)

*** Read processing                                               :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Demultiplexing and barcode removal
- Quality control / trimming

** General workflow (2/2)

*** /de novo/ assembly or mapping back                              :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Consensus sequences from /de novo/ assembly
- Mapping back the reads to consensus (or to reference genome)

*** Variant calling and allelotyping                                :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Variant calling (filtering? likelihood? bayesian?)
- Genotyping / allelotyping

*** Downstream analysis                                           :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- Genome scans
- QTL mapping
- Phylogenies
- etc...

** Nine-spined stickleback in Fenno-Scandia

*** Nine-spined stickleback                               :B_ignoreheading:
:PROPERTIES:
:BEAMER_env: ignoreheading
:END:

**** Nine-spined stickleback                               :B_block:BMCOL:
:PROPERTIES:
:BEAMER_env: block
:BEAMER_col: 0.5
:END:
- Versatile fish species
- Recent history of recolonization (Teacher 2011)
- Evidences of *local adaptation* (Prof. Merilä's group)

**** Map                                                           :BMCOL:
:PROPERTIES:
:BEAMER_col: 0.5
:END:
[[file:images/teacher-2011_map-pungitius.png]]

\pause

*** Local adaptation                                      :B_ignoreheading:
:PROPERTIES:
:BEAMER_env: ignoreheading
:END:

# **** Local adaptation                                      :B_block:BMCOL:
# :PROPERTIES:
# :BEAMER_env: block
# :BEAMER_col: 0.9
# :END:
# - Differences between freshwater and marine populations
# - Effect of community composition
# - Genetic-based differences in many traits

*** Giant ninespine                                       :B_ignoreheading:
:PROPERTIES:
:BEAMER_env: ignoreheading
:END:

**** Picture                                                       :BMCOL:
:PROPERTIES:
:BEAMER_col: 0.9
:END:
[[file:images/giant-9spine_EGRU-helsinki.png]]

** RAD tag experiments

*** Context and approach                                    :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.45
:BEAMER_env: block
:END:
- No transcriptomic or genomic resources
- But *three-spined stickleback genome* available
- Aim: *mapping the genetic differences* associated with local adaptation
\pause
- RAD tag approach
  + \alert{R}estriction-site-\alert{A}ssociated \alert{D}NA tag
  + Individuals pooled per populations
  + Discover SNPs and estimate *allele frequencies*

***                                                  :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.5
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/rad-tags.png]]
#+END_CENTER

** Results (1/2)

*** Low coverage                                          :B_ignoreheading:
:PROPERTIES:
:BEAMER_env: ignoreheading
:END:

**** Low coverage issues                                      :B_block:BMCOL:
:PROPERTIES:
:BEAMER_col: 0.6
:BEAMER_env: block
:END:
- SNP coverage lower than expected
- Populations *pooled by habitat type*

**** Map                                                           :BMCOL:
:PROPERTIES:
:BEAMER_col: 0.35
:END:
#+BEGIN_CENTER
#+LATEX_ATTR: :width 0.8\textwidth
[[file:images/map_9spines_bruneaux-2013.png]]
#+END_CENTER

\pause

*** Kernel smoothing and permutation tests                        :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
#+LATEX_ATTR: :width 0.9\textwidth
[[file:images/gst-profile-pval_bruneaux-2103.png]]
#+END_CENTER

** Results (2/2)

*** Identification of candidate genes
- Annotations from the three-spined stickleback genome
- Gene Ontology information

\pause

*** GO enrichment tests                                           :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
#+BEGIN_CENTER
[[file:images/high-gst-GO-bruneaux-2013.png]]
#+END_CENTER

** During the first part of the practicals

*** Simple scripts can be used also                               :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:
- This is one thing I want to show during the practical
- The objective is to get a *good grip* and a *good feeling/understanding* about
  the data with simple, straightforward methods
- Once we are comfortable, we can choose to apply *more complex methods* which
  rely on third-party scripts
- It is important to *understand what the third partyscripts do!*

* Attic                                                            :noexport:

** Typical analysis

- Depends on the question and objectives!
- Experimental/sampling design
- DNA extraction, library preps
- Barcoding? pooling?
- Sequencing (service?)
- Reads cleaning
- demultiplexing
- assembly or alignment (mapping if reference available)
- genotype calling
- RAD stops here (i.e. lots of markers and allele frequencies for different populations or genotypes for individuals)
- Downstream analysis:
  - genome scan
  - Fst, Gst, outlier detection
  - Phylogeography
  - Parallel evolution
  - and many other things...

** ddRAD
Examples, applications
(population genomics, mapping, QTL, phylogeography, ...)
or put those appplications in more general context?
nonmodel species, marker discovery
anything that needs lots of markers in lots of individuals or pools

** Uses of RAD tags

***                                                                 :BMCOL:
:PROPERTIES:
:BEAMER_col: 0.45
:END:

**** Population genomics                                         :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:

Hess 2013 - Pacific lamprey

**** Phylogeography                                              :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:

Emerson 2011

***                                                                 :BMCOL:
:PROPERTIES:
:BEAMER_col: 0.45
:END:

**** QTL mapping                                                 :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:

Houston 2012

**** Phylogenies                                                 :B_block:
:PROPERTIES:
:BEAMER_env: block
:END:

Rubin 2012

** Tour of other tools and specific analyses

To illustrate some specific points (e.g. likelihood or bayesian based genotyping
or allele frequency estimates or Fst calculations, ...)

* Org config                                                       :noexport:

#+OPTIONS: H:2 toc:nil
#+STARTUP: beamer
#+LaTeX_CLASS: beamer
#+LaTeX_CLASS_OPTIONS: [big]
#+LaTeX_HEADER: \usepackage{lmodern}
#+LaTeX_HEADER: \usetheme{Boadilla}
#+latex_header: \usecolortheme{whale}
#+LaTeX_HEADER: \setbeamertemplate{footline}{}
#+LaTeX_HEADER: \setbeamertemplate{navigation symbols}{}
#+LaTeX_HEADER: \setbeamertemplate{itemize items}[default]
#+LaTeX_HEADER: \setbeamertemplate{enumerate items}[circle]
#+LaTeX_HEADER: \setbeamertemplate{alert}{\textbf}
# http://tex.stackexchange.com/questions/171705/changing-your-bullet-points-in-beamer-block-maybe-boadilla
# http://tex.stackexchange.com/questions/68347/different-styles-of-bullets-of-enumerate
# http://tex.stackexchange.com/questions/66995/modify-footer-of-slides
# http://askubuntu.com/questions/98664/how-can-i-get-smooth-fonts-in-lyx
# http://orgmode.org/worg/exporters/beamer/tutorial.html
# http://orgmode.org/manual/Beamer-export.html
# https://lists.gnu.org/archive/html/emacs-orgmode/2008-07/msg00163.html
