<html>
<head>
<meta name="Title" content="Bibliography notes" />
<meta name="HTML_HEADER" content="<META NAME="Summary" CONTENT="bibliographyNotes">" />
<meta name="URL" content="bibliography-notes.html" />
<meta name="Save_as" content="bibliography-notes.html" />
<meta name="Sortorder" content="020" />
<meta name="Slug" content="bibliographyNotes" />
</head>
<body>
  <div id="table-of-contents">
   <h2>
    Table of Contents
   </h2>
   <div id="text-table-of-contents">
    <ul>
     <li>
      <a href="#sec-1">
       1.
       <b>
        Miller et al. (2007) - Genome Research
       </b>
       (link)
       <span class="tag">
        <span class="major">
         major
        </span>
       </span>
      </a>
     </li>
     <li>
      <a href="#sec-2">
       2.
       <b>
        Miller et al. (2007) - Genome Biology
       </b>
       (link)
       <span class="tag">
        <span class="major">
         major
        </span>
       </span>
      </a>
      <ul>
       <li>
        <a href="#sec-2-1">
         2.1.
         <span class="todo TODO">
          TODO
         </span>
         Read in more details
        </a>
       </li>
      </ul>
     </li>
     <li>
      <a href="#sec-3">
       3. Lewis et al. (2007) - Genetics (link)
      </a>
     </li>
     <li>
      <a href="#sec-4">
       4.
       <b>
        Baird et al. (2008) - PLoS ONE
       </b>
       (link)
       <span class="tag">
        <span class="major">
         major
        </span>
       </span>
      </a>
     </li>
     <li>
      <a href="#sec-5">
       5.
       <b>
        Hohenlohe et al. (2010) - PLoS Genetics
       </b>
       (link)
       <span class="tag">
        <span class="major">
         major
        </span>
       </span>
      </a>
     </li>
     <li>
      <a href="#sec-6">
       6.
       <b>
        Hohenlohe et al. (2011) - Molecular Ecology Resources
       </b>
       (link)
       <span class="tag">
        <span class="major">
         major
        </span>
       </span>
      </a>
     </li>
     <li>
      <a href="#sec-7">
       7. Etter et al. (2011) - PLoS ONE (link)
      </a>
     </li>
    </ul>
   </div>
  </div>
  <p>
   The idea of using restriction site associated DNA (RAD) markers on a large
scale was first introduced in a paper by
   <a href="http://genome.cshlp.org/content/17/2/240.long">
    Miller et al. (2007)
   </a>
   from Eric
A. Johnson's group.
  </p>
  <p>
   There are important differences between methods based on
   <b>
    restriction site
polymorphism
   </b>
   (like
   <a href="http://genome.cshlp.org/content/17/2/240.long">
    Miller et al. 2007
   </a>
   ) and methods that use restriction sites
as a way to
   <b>
    decrease genomic complexity
   </b>
   before sequencing.
  </p>
  <p>
   Below are bibliography notes related to the use of RAD tags.
  </p>
  <div class="outline-2" id="outline-container-sec-1">
   <h2 id="sec-1">
    <span class="section-number-2">
     1
    </span>
    <b>
     Miller et al. (2007) - Genome Research
    </b>
    (
    <a href="http://genome.cshlp.org/content/17/2/240.long">
     link
    </a>
    )
    <span class="tag">
     <span class="major">
      major
     </span>
    </span>
   </h2>
   <div class="outline-text-2" id="text-1">
    <p>
     <b>
      Rapid and cost-effective polymorphism identification and genotyping using
restriction site associated DNA (RAD) markers
     </b>
    </p>
    <ul class="org-ul">
     <li>
      Summary
      <ul class="org-ul">
       <li>
        Eric A. Johnson's group at the University of Oregon
       </li>
       <li>
        Definition of RAD tags: "Restriction site associated DNA (RAD) tags are a
        <b>
         genome-wide representation of every site of a particular restriction enzyme
by short DNA tags
        </b>
        "
       </li>
       <li>
        Here, looking for polymorphism that disrupt restriction sites (like RFLP,
AFLP) but for many more loci than the previous methods.
       </li>
       <li>
        Not sequencing but use of a micro-array platform
       </li>
       <li>
        "
        <b>
         RAD markers [â€¦] allow high-throughput, high-resolution genotyping in
both model and nonmodel systems
        </b>
        "
       </li>
      </ul>
     </li>
     <li>
      Interest of RAD tags
      <ul class="org-ul">
       <li>
        Genetic mapping (recombination mapping, association mapping) needs
"high-throughput, cost-effective methods to identify and genotype
polymorphisms"
       </li>
       <li>
        Previous methods:
        <ul class="org-ul">
         <li>
          allele-specific oligonucleotide microarrays (thousands of SNPs scored per
hybridization)
         </li>
         <li>
          direct hybridization of labelled genomic DNA to short oligonucleotide
arrays
         </li>
         <li>
          single feature polymorphisms (SFP) can be discovered and typed in a similar
way
         </li>
        </ul>
       </li>
       <li>
        Those methods need costly preliminary steps (sequencing, oligonucleotide
synthesis) and are not easily feasible for non-model organisms
       </li>
      </ul>
     </li>
     <li>
      Principle
      <ul class="org-ul">
       <li>
        Very good illustration in
        <a href="http://genome.cshlp.org/content/17/2/240.long#F1">
         Figure 1
        </a>
        .
       </li>
       <li>
        Digest - link tags - shear DNA - purify tags - release tags - type tags
       </li>
      </ul>
     </li>
     <li>
      Demonstration
      <ul class="org-ul">
       <li>
        Mapping of a breakpoint on a
        <i>
         Drosophila
        </i>
        chromosome
       </li>
       <li>
        Identification of the major lateral plate locus in three-spine stickleback
       </li>
      </ul>
     </li>
     <li>
      Discussion
      <ul class="org-ul">
       <li>
        Here RAD markers use the
        <b>
         restriction site as the polymorphic marker
        </b>
        (no
sequencing)
       </li>
       <li>
        Easy to produce microarray type of genotyping resource
       </li>
       <li>
        Bulk genotyping is possible
       </li>
       <li>
        If markers location is determined, genetic mapping is possible
       </li>
      </ul>
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-2">
   <h2 id="sec-2">
    <span class="section-number-2">
     2
    </span>
    <b>
     Miller et al. (2007) - Genome Biology
    </b>
    (
    <a href="http://genomebiology.com/content/8/6/R105">
     link
    </a>
    )
    <span class="tag">
     <span class="major">
      major
     </span>
    </span>
   </h2>
   <div class="outline-text-2" id="text-2">
    <p>
     <b>
      RAD marker microarrays enable rapid mapping of zebrafish mutations
     </b>
    </p>
    <ul class="org-ul">
     <li>
      Developed a RAD marker microarray resource for the zebrafish
     </li>
     <li>
      Bulk segregant approach to map mutations
     </li>
     <li>
      Assay for individual RAD markers in pooled populations
     </li>
    </ul>
   </div>
   <div class="outline-3" id="outline-container-sec-2-1">
    <h3 id="sec-2-1">
     <span class="section-number-3">
      2.1
     </span>
     <span class="todo TODO">
      TODO
     </span>
     Read in more details
    </h3>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-3">
   <h2 id="sec-3">
    <span class="section-number-2">
     3
    </span>
    Lewis et al. (2007) - Genetics (
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2034621/">
     link
    </a>
    )
   </h2>
   <div class="outline-text-2" id="text-3">
    <p>
     <b>
      High-Density Detection of Restriction-Site-Associated DNA Markers for Rapid
Mapping of Mutated Loci in Neurospora
     </b>
    </p>
    <ul class="org-ul">
     <li>
      Micro-array based RAD markers in
      <i>
       Neurospora crassa
      </i>
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-4">
   <h2 id="sec-4">
    <span class="section-number-2">
     4
    </span>
    <b>
     Baird et al. (2008) - PLoS ONE
    </b>
    (
    <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0003376">
     link
    </a>
    )
    <span class="tag">
     <span class="major">
      major
     </span>
    </span>
   </h2>
   <div class="outline-text-2" id="text-4">
    <p>
     <b>
      Rapid SNP Discovery and Genetic Mapping Using Sequenced RAD Markers
     </b>
    </p>
    <ul class="org-ul">
     <li>
      RAD tags associated with Illumina sequencing for SNP discovery
     </li>
     <li>
      SNP identified by sequenced polymorphism and disrupted restriction sites
     </li>
     <li>
      Marker density can be tuned by the choice of the enzymes (and their number)
     </li>
     <li>
      Samples can be multiplexed by barcoding
     </li>
     <li>
      Useful with or without a reference genome
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-5">
   <h2 id="sec-5">
    <span class="section-number-2">
     5
    </span>
    <b>
     Hohenlohe et al. (2010) - PLoS Genetics
    </b>
    (
    <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">
     link
    </a>
    )
    <span class="tag">
     <span class="major">
      major
     </span>
    </span>
   </h2>
   <div class="outline-text-2" id="text-5">
    <p>
     <b>
      Population Genomics of Parallel Adaptation in Threespine Stickleback using
Sequenced RAD Tags
     </b>
    </p>
    <p>
     A major paper about sequenced RAD tags.
    </p>
    <ul class="org-ul">
     <li>
      Use natural populations, not lab crosses
     </li>
     <li>
      Genome scan of nucleotide diversity and differentiation in 5 natural
populations of threespine stickleback
     </li>
     <li>
      Illumina-sequenced RAD tags
     </li>
     <li>
      Identify and type more than 45000 SNPs in each of 100 individuals
     </li>
     <li>
      Genomic regions with balancing and divergent selection, consistent across
populations
     </li>
     <li>
      Identify the genes in regions of interest (regions of parallel
differentiation across independent populations)
     </li>
     <li>
      Methods:
      <ul class="org-ul">
       <li>
        Genome is available
       </li>
       <li>
        Barcoded sample
       </li>
       <li>
        Kernel smoothing and permutation testing
       </li>
      </ul>
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-6">
   <h2 id="sec-6">
    <span class="section-number-2">
     6
    </span>
    <b>
     Hohenlohe et al. (2011) - Molecular Ecology Resources
    </b>
    (
    <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1755-0998.2010.02967.x/abstract">
     link
    </a>
    )
    <span class="tag">
     <span class="major">
      major
     </span>
    </span>
   </h2>
   <div class="outline-text-2" id="text-6">
    <p>
     <b>
      Next-generation RAD sequencing identifies thousands of SNPs for assessing
 hybridization between rainbow and westslope cutthroat trout
     </b>
    </p>
    <ul class="org-ul">
     <li>
      Nonmodel organism (no genome available)
     </li>
     <li>
      Salmonid fish, several whole-genome duplication
     </li>
     <li>
      RAD sequencing to identify SNP loci with fixed allelic differences between
introduced rainbow trout and native westslope cutthroat trout
     </li>
     <li>
      Use 24 barcode-labelled individuals
     </li>
     <li>
      Use Stacks for analysis (genotyping)
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-7">
   <h2 id="sec-7">
    <span class="section-number-2">
     7
    </span>
    Etter et al. (2011) - PLoS ONE (
    <a href="http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0018561">
     link
    </a>
    )
   </h2>
   <div class="outline-text-2" id="text-7">
    <p>
     <b>
      Local De Novo Assembly of RAD Paired-End Contigs Using Short Sequencing Reads
     </b>
    </p>
    <ul class="org-ul">
     <li>
      Use paired-end sequencing of RAD fragments to build contigs on the side which
is cut by random shearing
     </li>
     <li>
      Identify SNP and determine haplotype in threespine stickleback
     </li>
     <li>
      Produce overlapping contigs of several hundred nucleotides in
      <i>
       E. coli
      </i>
      and
threespine stickleback
     </li>
     <li>
      A circularization step can allow a local assembly of up to 5 kb
     </li>
    </ul>
   </div>
  </div>
 </div>
 <div class="status" id="postamble">
  <p class="author">
   Author: Matthieu Bruneaux
  </p>
  <p class="date">
   Created: 2015-02-26 Thu 17:41
  </p>
  <p class="creator">
   <a href="http://www.gnu.org/software/emacs/">
    Emacs
   </a>
   24.3.50.1 (
   <a href="http://orgmode.org">
    Org
   </a>
   mode 8.2.3a)
  </p>
  <p class="validation">
   <a href="http://validator.w3.org/check?uri=referer">
    Validate
   </a>
  </p>
</body>
</html>
