<html>
<head>
<meta name="Title" content="Draft notes" />
<meta name="Summary" content="draftNotes" />
<meta name="URL" content="draft-notes.html" />
<meta name="Save_as" content="draft-notes.html" />
<meta name="Sortorder" content="030" />
<meta name="Slug" content="draftNotes" />
</head>
<body>
  <div id="table-of-contents">
   <h2>
    Table of Contents
   </h2>
   <div id="text-table-of-contents">
    <ul>
     <li>
      <a href="#sec-1">
       1. Course about RAD-tag sequencing - Roscoff 2015
      </a>
      <ul>
       <li>
        <a href="#sec-1-1">
         1.1. Problematic
        </a>
       </li>
       <li>
        <a href="#sec-1-2">
         1.2. Presentation of the RAD-tag approach (history)
        </a>
       </li>
       <li>
        <a href="#sec-1-3">
         1.3. Use of RAD tags
        </a>
       </li>
      </ul>
     </li>
     <li>
      <a href="#sec-2">
       2. Practicals
      </a>
      <ul>
       <li>
        <a href="#sec-2-1">
         2.1. Teaching objectives
        </a>
       </li>
       <li>
        <a href="#sec-2-2">
         2.2. Available data
        </a>
       </li>
      </ul>
     </li>
     <li>
      <a href="#sec-3">
       3. Important people and groups
      </a>
      <ul>
       <li>
        <a href="#sec-3-1">
         3.1. Three central groups
        </a>
       </li>
       <li>
        <a href="#sec-3-2">
         3.2. People
        </a>
       </li>
      </ul>
     </li>
     <li>
      <a href="#sec-4">
       4. Other interesting papers
      </a>
     </li>
    </ul>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-1">
   <h2 id="sec-1">
    <span class="section-number-2">
     1
    </span>
    Course about RAD-tag sequencing - Roscoff 2015
   </h2>
   <div class="outline-text-2" id="text-1">
   </div>
   <div class="outline-3" id="outline-container-sec-1-1">
    <h3 id="sec-1-1">
     <span class="section-number-3">
      1.1
     </span>
     Problematic
    </h3>
    <div class="outline-text-3" id="text-1-1">
     <ul class="org-ul">
      <li>
       High coverage
      </li>
      <li>
       Genome reduction
      </li>
      <li>
       Non-model species
      </li>
     </ul>
    </div>
   </div>
   <div class="outline-3" id="outline-container-sec-1-2">
    <h3 id="sec-1-2">
     <span class="section-number-3">
      1.2
     </span>
     Presentation of the RAD-tag approach (history)
    </h3>
    <div class="outline-text-3" id="text-1-2">
     <ul class="org-ul">
      <li>
       Protocol
      </li>
      <li>
       Single or double digest
      </li>
      <li>
       Single end or paired ends
      </li>
     </ul>
    </div>
   </div>
   <div class="outline-3" id="outline-container-sec-1-3">
    <h3 id="sec-1-3">
     <span class="section-number-3">
      1.3
     </span>
     Use of RAD tags
    </h3>
    <div class="outline-text-3" id="text-1-3">
     <ul class="org-ul">
      <li>
       SNP discovery
      </li>
      <li>
       Population genetics
      </li>
      <li>
       Functional genomics
      </li>
     </ul>
    </div>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-2">
   <h2 id="sec-2">
    <span class="section-number-2">
     2
    </span>
    Practicals
   </h2>
   <div class="outline-text-2" id="text-2">
   </div>
   <div class="outline-3" id="outline-container-sec-2-1">
    <h3 id="sec-2-1">
     <span class="section-number-3">
      2.1
     </span>
     Teaching objectives
    </h3>
    <div class="outline-text-3" id="text-2-1">
     <ul class="org-ul">
      <li>
       Sequence quality check: trimming, quality control
      </li>
      <li>
       Stack building (STACKS or other tools, with or without paired-ends
information)
      </li>
      <li>
       SNP calling and genotyping (VCFtools)
      </li>
      <li>
       Allele frequency estimates (individuals or pools?)
      </li>
      <li>
       Population genetics: comparison of microsats and SNPs trees (cf. Anti)
      </li>
      <li>
       Population genomics: localisation, genome profiling and Gst calculation,
kernel smoothing
      </li>
      <li>
       Functional analysis (outside the scope of RAD-tags
       <i>
        sensus stricto
       </i>
       )
      </li>
      <li>
       Predicting the number of fragmnts based on published genome sequence and
restriction sites with python scripts
      </li>
     </ul>
    </div>
   </div>
   <div class="outline-3" id="outline-container-sec-2-2">
    <h3 id="sec-2-2">
     <span class="section-number-3">
      2.2
     </span>
     Available data
    </h3>
    <div class="outline-text-3" id="text-2-2">
     <ul class="org-ul">
      <li>
       Baird 2008: short reads for three-spines and
       <i>
        Neurospora crassa
       </i>
      </li>
      <li>
       Hohenlohe 2010: short reads for three-spines (population genomics of parallel
evolution)
      </li>
      <li>
       Hohenlohe 2011: short reads for rainbow trout and westslope cutthroat trout
(SNP identification for hybridization diagnostic)
      </li>
     </ul>
    </div>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-3">
   <h2 id="sec-3">
    <span class="section-number-2">
     3
    </span>
    Important people and groups
   </h2>
   <div class="outline-text-2" id="text-3">
   </div>
   <div class="outline-3" id="outline-container-sec-3-1">
    <h3 id="sec-3-1">
     <span class="section-number-3">
      3.1
     </span>
     Three central groups
    </h3>
    <div class="outline-text-3" id="text-3-1">
     <p>
      To the best of my knowledge, there are three groups which are involved in the
origin of the RAD tags method and which have produced a lot of the initial
papers. They are also producing a lot of the software that can be used for RAD
data analysis. The groups are:
     </p>
     <ul class="org-ul">
      <li>
       Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon):
       <a href="http://molbio.uoregon.edu/johnson/">
        link
       </a>
      </li>
      <li>
       William A. Cresko's group (Institute of Ecology and Evolution, U. Oregon):
       <a href="http://creskolab.uoregon.edu/">
        link
       </a>
       . The Cresko lab is responsible for the
       <a href="http://creskolab.uoregon.edu/stacks/">
        Stacks
       </a>
       pipeline that can be used
to analyse RAD data.
      </li>
      <li>
       Paul A. Hohenlohe (Depts of Biological Science and Statistics, U. Idaho):
       <a href="http://webpages.uidaho.edu/hohenlohe/index.html">
        link
       </a>
       . They also have some software and their news page has some interesting
information about what's going on with RAD.
      </li>
     </ul>
    </div>
   </div>
   <div class="outline-3" id="outline-container-sec-3-2">
    <h3 id="sec-3-2">
     <span class="section-number-3">
      3.2
     </span>
     People
    </h3>
    <div class="outline-text-3" id="text-3-2">
     <ul class="org-ul">
      <li>
       Miller Michael R.
      </li>
      <li>
       Johnson Eric A.
      </li>
      <li>
       Cresko William A.
      </li>
      <li>
       Baird Nathan A.
      </li>
      <li>
       Catchen Julian M.
      </li>
     </ul>
    </div>
   </div>
  </div>
  <div class="outline-2" id="outline-container-sec-4">
   <h2 id="sec-4">
    <span class="section-number-2">
     4
    </span>
    Other interesting papers
   </h2>
   <div class="outline-text-2" id="text-4">
    <ul class="org-ul">
     <li>
      Hohenlohe 2010, "Using population genomics to detect selection in natural
populations: key concepts and methodological considerations" (
      <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/">
       link
      </a>
      ).
An interesting review about population genomics.
     </li>
    </ul>
   </div>
  </div>
 </div>
 <div class="status" id="postamble">
  <p class="author">
   Author: Matthieu Bruneaux
  </p>
  <p class="date">
   Created: 2015-02-26 Thu 17:41
  </p>
  <p class="creator">
   <a href="http://www.gnu.org/software/emacs/">
    Emacs
   </a>
   24.3.50.1 (
   <a href="http://orgmode.org">
    Org
   </a>
   mode 8.2.3a)
  </p>
  <p class="validation">
   <a href="http://validator.w3.org/check?uri=referer">
    Validate
   </a>
  </p>
</body>
</html>
