Title: Draft notes
Summary: draftNotes
URL: draft-notes.html
Save_as: draft-notes.html
Sortorder: 030
Slug: draftNotes

Course about RAD-tag sequencing - Roscoff 2015
==============================================

Problematic
-----------

-   High coverage
-   Genome reduction
-   Non-model species

Presentation of the RAD-tag approach (history)
----------------------------------------------

-   Protocol
-   Single or double digest
-   Single end or paired ends

Use of RAD tags
---------------

-   SNP discovery
-   Population genetics
-   Functional genomics

\#
Practicals
==========

Teaching objectives
-------------------

-   Sequence quality check: trimming, quality control
-   Stack building (STACKS or other tools, with or without paired-ends
    information)
-   SNP calling and genotyping (VCFtools)
-   Allele frequency estimates (individuals or pools?)
-   Population genetics: comparison of microsats and SNPs trees (cf.
    Anti)
-   Population genomics: localisation, genome profiling and Gst
    calculation, kernel smoothing
-   Functional analysis (outside the scope of RAD-tags *sensus stricto*)
-   Predicting the number of fragmnts based on published genome sequence
    and restriction sites with python scripts

Available data
--------------

-   Baird 2008: short reads for three-spines and *Neurospora crassa*
-   Hohenlohe 2010: short reads for three-spines (population genomics of
    parallel evolution)
-   Hohenlohe 2011: short reads for rainbow trout and westslope
    cutthroat trout (SNP identification for hybridization diagnostic)

-   Important people and groups

Three central groups
--------------------

To the best of my knowledge, there are three groups which are involved
in the origin of the RAD tags method and which have produced a lot of
the initial papers. They are also producing a lot of the software that
can be used for RAD data analysis. The groups are:
-   Eric A. Johnson's group (Institute of Molecular Biology, U. Oregon):
    [link](http://molbio.uoregon.edu/johnson/)
-   William A. Cresko's group (Institute of Ecology and Evolution, U.
    Oregon): [link](http://creskolab.uoregon.edu/). The Cresko lab is
    responsible for the [Stacks](http://creskolab.uoregon.edu/stacks/)
    pipeline that can be used to analyse RAD data.
-   Paul A. Hohenlohe (Depts of Biological Science and Statistics, U.
    Idaho): [link](http://webpages.uidaho.edu/hohenlohe/index.html).
    They also have some software and their news page has some
    interesting information about what's going on with RAD.

People
------

-   Miller Michael R.
-   Johnson Eric A.
-   Cresko William A.
-   Baird Nathan A.
-   Catchen Julian M.

-   Other interesting papers

-   Hohenlohe 2010, "Using population genomics to detect selection in
    natural populations: key concepts and methodological considerations"
    ([link](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3016716/)). An
    interesting review about population genomics.


