<html>
<head>
<meta name="Title" content="Downstream analysis - Complete workflow" />
<meta name="Summary" content="Downstream analysis" />
<meta name="URL" content="part-one-05-downstream-analysis.html" />
<meta name="Save_as" content="part-one-05-downstream-analysis.html" />
<meta name="Status" content="hidden" />
</head>
<body>


<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1">Downstream analysis</h2>
<div class="outline-text-2" id="text-1">
<p>
<div class="navLink"><a href="part-one-04-variant-calling-genotyping.html">Previous: Variant calling and genotyping</a></div>
</p>
<p>
At this point the basic steps which are common to many RAD-seq experiments have been
performed:
</p>
<ul class="org-ul">
<li>raw reads processing (filtering and trimming)
</li>
<li>de novo assembly (or mapping back to a reference if available)
</li>
<li>reads mapping back to the consensus sequences
</li>
<li>variant calling
</li>
<li>genotyping and allele frequency estimates
</li>
</ul>
<p>
We are ready to perform the downstream analyses which will answer the
biological question(s) that led us to perform the RAD-seq in the first
place. Those analyses includes population genomics, phylogeography,
phylogenetics, and others.
</p>
<p>
In this practical the downstream analyses will be:
</p>
<ol class="org-ol">
<li>Building a phylogeographic tree of the populations (not demonstrated)
</li>
<li>Mapping back the consensus sequences to a reference genome
</li>
<li>Predicting the synonymous or non-synonymous effect of mutations
</li>
<li>Performing a genome scan using G<sub>st</sub> to detect regions of low/high
differentiation
</li>
<li>Test for gene-ontology enrichment in those regions
</li>
</ol>
<p>
<div class="navLink"><a href="part-one-06-phylogeographic-tree.html">Next: Phylogeographic tree</a></div>
</p>
</div>
</div>
</body></html>
