<html>
<head>
<meta name="Title" content="De novo assembly - Complete workflow" />
<meta name="Summary" content="Assembly" />
<meta name="URL" content="part-one-03-de-novo-assembly.html" />
<meta name="Save_as" content="part-one-03-de-novo-assembly.html" />
<meta name="Status" content="hidden" />
</head>
<body>


<div class="outline-2" id="outline-container-sec-1">
<h2 id="sec-1"><i>De novo</i> assembly</h2>
<div class="outline-text-2" id="text-1">
<p>
<div class="navLink"><a href="part-one-02-raw-reads-processing.html">Previous: Raw reads processing</a></div>
</p>
</div>
<div class="outline-3" id="outline-container-sec-1-1">
<h3 id="sec-1-1">Assembly into contigs</h3>
<div class="outline-text-3" id="text-1-1">
<p>
We choose to go for a de novo assembly. Several tools were tested, in the end
we went for an older tool, CAP3. Discuss the reasons with Susie?
</p>
<p>
An alternative approach would have been to map the reads back to the reference
genome. However, when mapping back we have to be quite stringent and allow only
for example one or two mismatch, and one unique good quality mapping
location. With the reference genome of a different species, this can be
problematic and the mapping back might be less efficient that creating
consensus contigs or stacks.
</p>
</div>
</div>
<div class="outline-3" id="outline-container-sec-1-2">
<h3 id="sec-1-2">Removal of repeat sequences</h3>
<div class="outline-text-3" id="text-1-2">
<p>
<div class="navLink"><a href="part-one-04-variant-calling-genotyping.html">Next: Variant calling and genotyping</a></div>
</p>
</div>
</div>
</div>
</body></html>
